{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "ac3a45f4",
   "metadata": {},
   "source": [
    "## Setup"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d36eb997",
   "metadata": {},
   "source": [
    "Load packages and set working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2d4e74a8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: graph\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Loading required package: GO.db\n",
      "\n",
      "Loading required package: AnnotationDbi\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Loading required package: SparseM\n",
      "\n",
      "\n",
      "Attaching package: ‘SparseM’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    backsolve\n",
      "\n",
      "\n",
      "\n",
      "groupGOTerms: \tGOBPTerm, GOMFTerm, GOCCTerm environments built.\n",
      "\n",
      "\n",
      "Attaching package: ‘topGO’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    members\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(topGO)\n",
    "library(ggplot2)\n",
    "library(tidyr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5aa2efa6",
   "metadata": {},
   "outputs": [],
   "source": [
    "setwd(\"/Users/bsierieb/Work-local/project_insulin/2022.04.06_RNA-seq_GO/data\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "651f4a12",
   "metadata": {},
   "source": [
    "Open `Table_S1_DE_genes.xlsx` in Excel and save each tab as a separate .csv file. For example, save the \"Fb_W_vs_G\" tab as `Table_S1_DE_genes_Fb_W_vs_G.csv`."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3bd95b23",
   "metadata": {},
   "source": [
    "Import the lists of DE genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "dab0a04f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>genes</th><th scope=col>logFC</th><th scope=col>logCPM</th><th scope=col>PValue</th><th scope=col>FDR</th><th scope=col>Annotation</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>LOC105191040</td><td>-9.456765</td><td>8.628889</td><td>2.19e-73</td><td>2.78e-69</td><td>fatty acid synthase                                </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>LOC112590176</td><td>-8.443456</td><td>7.072905</td><td>6.81e-64</td><td>4.31e-60</td><td>fatty acid synthase-like                           </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>LOC109504628</td><td>-6.877841</td><td>7.878917</td><td>4.53e-56</td><td>1.91e-52</td><td>cytochrome P450 6B5-like                           </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>LOC112590521</td><td>-9.123070</td><td>9.022293</td><td>6.59e-55</td><td>2.09e-51</td><td>putative fatty acyl-CoA reductase CG5065 isoform X1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>LOC105191801</td><td>-8.094335</td><td>9.139217</td><td>5.17e-53</td><td>1.31e-49</td><td>acyl-CoA Delta(11) desaturase                      </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>LOC112588736</td><td>-8.550090</td><td>5.215000</td><td>4.33e-52</td><td>9.14e-49</td><td>fatty acid synthase-like                           </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & genes & logFC & logCPM & PValue & FDR & Annotation\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & LOC105191040 & -9.456765 & 8.628889 & 2.19e-73 & 2.78e-69 & fatty acid synthase                                \\\\\n",
       "\t2 & LOC112590176 & -8.443456 & 7.072905 & 6.81e-64 & 4.31e-60 & fatty acid synthase-like                           \\\\\n",
       "\t3 & LOC109504628 & -6.877841 & 7.878917 & 4.53e-56 & 1.91e-52 & cytochrome P450 6B5-like                           \\\\\n",
       "\t4 & LOC112590521 & -9.123070 & 9.022293 & 6.59e-55 & 2.09e-51 & putative fatty acyl-CoA reductase CG5065 isoform X1\\\\\n",
       "\t5 & LOC105191801 & -8.094335 & 9.139217 & 5.17e-53 & 1.31e-49 & acyl-CoA Delta(11) desaturase                      \\\\\n",
       "\t6 & LOC112588736 & -8.550090 & 5.215000 & 4.33e-52 & 9.14e-49 & fatty acid synthase-like                           \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | genes &lt;chr&gt; | logFC &lt;dbl&gt; | logCPM &lt;dbl&gt; | PValue &lt;dbl&gt; | FDR &lt;dbl&gt; | Annotation &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | LOC105191040 | -9.456765 | 8.628889 | 2.19e-73 | 2.78e-69 | fatty acid synthase                                 |\n",
       "| 2 | LOC112590176 | -8.443456 | 7.072905 | 6.81e-64 | 4.31e-60 | fatty acid synthase-like                            |\n",
       "| 3 | LOC109504628 | -6.877841 | 7.878917 | 4.53e-56 | 1.91e-52 | cytochrome P450 6B5-like                            |\n",
       "| 4 | LOC112590521 | -9.123070 | 9.022293 | 6.59e-55 | 2.09e-51 | putative fatty acyl-CoA reductase CG5065 isoform X1 |\n",
       "| 5 | LOC105191801 | -8.094335 | 9.139217 | 5.17e-53 | 1.31e-49 | acyl-CoA Delta(11) desaturase                       |\n",
       "| 6 | LOC112588736 | -8.550090 | 5.215000 | 4.33e-52 | 9.14e-49 | fatty acid synthase-like                            |\n",
       "\n"
      ],
      "text/plain": [
       "  genes        logFC     logCPM   PValue   FDR     \n",
       "1 LOC105191040 -9.456765 8.628889 2.19e-73 2.78e-69\n",
       "2 LOC112590176 -8.443456 7.072905 6.81e-64 4.31e-60\n",
       "3 LOC109504628 -6.877841 7.878917 4.53e-56 1.91e-52\n",
       "4 LOC112590521 -9.123070 9.022293 6.59e-55 2.09e-51\n",
       "5 LOC105191801 -8.094335 9.139217 5.17e-53 1.31e-49\n",
       "6 LOC112588736 -8.550090 5.215000 4.33e-52 9.14e-49\n",
       "  Annotation                                         \n",
       "1 fatty acid synthase                                \n",
       "2 fatty acid synthase-like                           \n",
       "3 cytochrome P450 6B5-like                           \n",
       "4 putative fatty acyl-CoA reductase CG5065 isoform X1\n",
       "5 acyl-CoA Delta(11) desaturase                      \n",
       "6 fatty acid synthase-like                           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>genes</th><th scope=col>logFC</th><th scope=col>logCPM</th><th scope=col>PValue</th><th scope=col>FDR</th><th scope=col>Annotation</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>12649</th><td>LOC105183414</td><td>2.177012</td><td>-1.309972</td><td>1</td><td>1</td><td>vesicular glutamate transporter 1                    </td></tr>\n",
       "\t<tr><th scope=row>12650</th><td>LOC112589917</td><td>2.177111</td><td>-1.309972</td><td>1</td><td>1</td><td>uncharacterized protein LOC112589917                 </td></tr>\n",
       "\t<tr><th scope=row>12651</th><td>LOC105189833</td><td>2.177120</td><td>-1.309972</td><td>1</td><td>1</td><td>myb/SANT-like DNA-binding domain-containing protein 3</td></tr>\n",
       "\t<tr><th scope=row>12652</th><td>LOC112589632</td><td>2.177135</td><td>-1.309972</td><td>1</td><td>1</td><td>uncharacterized protein LOC112589632                 </td></tr>\n",
       "\t<tr><th scope=row>12653</th><td>LOC112590213</td><td>2.177206</td><td>-1.309972</td><td>1</td><td>1</td><td>serine/arginine repetitive matrix protein 1-like     </td></tr>\n",
       "\t<tr><th scope=row>12654</th><td>LOC112590430</td><td>2.177327</td><td>-1.309972</td><td>1</td><td>1</td><td>histone-lysine N-methyltransferase SETMAR-like       </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & genes & logFC & logCPM & PValue & FDR & Annotation\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t12649 & LOC105183414 & 2.177012 & -1.309972 & 1 & 1 & vesicular glutamate transporter 1                    \\\\\n",
       "\t12650 & LOC112589917 & 2.177111 & -1.309972 & 1 & 1 & uncharacterized protein LOC112589917                 \\\\\n",
       "\t12651 & LOC105189833 & 2.177120 & -1.309972 & 1 & 1 & myb/SANT-like DNA-binding domain-containing protein 3\\\\\n",
       "\t12652 & LOC112589632 & 2.177135 & -1.309972 & 1 & 1 & uncharacterized protein LOC112589632                 \\\\\n",
       "\t12653 & LOC112590213 & 2.177206 & -1.309972 & 1 & 1 & serine/arginine repetitive matrix protein 1-like     \\\\\n",
       "\t12654 & LOC112590430 & 2.177327 & -1.309972 & 1 & 1 & histone-lysine N-methyltransferase SETMAR-like       \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | genes &lt;chr&gt; | logFC &lt;dbl&gt; | logCPM &lt;dbl&gt; | PValue &lt;dbl&gt; | FDR &lt;dbl&gt; | Annotation &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 12649 | LOC105183414 | 2.177012 | -1.309972 | 1 | 1 | vesicular glutamate transporter 1                     |\n",
       "| 12650 | LOC112589917 | 2.177111 | -1.309972 | 1 | 1 | uncharacterized protein LOC112589917                  |\n",
       "| 12651 | LOC105189833 | 2.177120 | -1.309972 | 1 | 1 | myb/SANT-like DNA-binding domain-containing protein 3 |\n",
       "| 12652 | LOC112589632 | 2.177135 | -1.309972 | 1 | 1 | uncharacterized protein LOC112589632                  |\n",
       "| 12653 | LOC112590213 | 2.177206 | -1.309972 | 1 | 1 | serine/arginine repetitive matrix protein 1-like      |\n",
       "| 12654 | LOC112590430 | 2.177327 | -1.309972 | 1 | 1 | histone-lysine N-methyltransferase SETMAR-like        |\n",
       "\n"
      ],
      "text/plain": [
       "      genes        logFC    logCPM    PValue FDR\n",
       "12649 LOC105183414 2.177012 -1.309972 1      1  \n",
       "12650 LOC112589917 2.177111 -1.309972 1      1  \n",
       "12651 LOC105189833 2.177120 -1.309972 1      1  \n",
       "12652 LOC112589632 2.177135 -1.309972 1      1  \n",
       "12653 LOC112590213 2.177206 -1.309972 1      1  \n",
       "12654 LOC112590430 2.177327 -1.309972 1      1  \n",
       "      Annotation                                           \n",
       "12649 vesicular glutamate transporter 1                    \n",
       "12650 uncharacterized protein LOC112589917                 \n",
       "12651 myb/SANT-like DNA-binding domain-containing protein 3\n",
       "12652 uncharacterized protein LOC112589632                 \n",
       "12653 serine/arginine repetitive matrix protein 1-like     \n",
       "12654 histone-lysine N-methyltransferase SETMAR-like       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Fb_W_vs_G <- read.csv('Table_S1_DE_genes_Fb_W_vs_G.csv')\n",
    "head(Fb_W_vs_G) # check\n",
    "tail(Fb_W_vs_G) # check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "17d2705d",
   "metadata": {},
   "outputs": [],
   "source": [
    "Fb_G_vs_R <- read.csv('Table_S1_DE_genes_Fb_G_vs_R.csv')\n",
    "Ov_W_vs_G <- read.csv('Table_S1_DE_genes_Ov_W_vs_G.csv')\n",
    "Ov_G_vs_R <- read.csv('Table_S1_DE_genes_Ov_G_vs_R.csv')\n",
    "cBr_W_vs_G <- read.csv('Table_S1_DE_genes_cBr_W_vs_G.csv')\n",
    "cBr_G_vs_R <- read.csv('Table_S1_DE_genes_cBr_G_vs_R.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "635ecd9a",
   "metadata": {},
   "source": [
    "Import gene:GO mappings generated by Karl Glastad using Blast2GO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "9fe62faa",
   "metadata": {},
   "outputs": [],
   "source": [
    "# since Karl's file has the topGO-compatible structure:\n",
    "# gene_ID<TAB>GO_ID1, GO_ID2, GO_ID3\n",
    "# readMappings() can be used to import it\n",
    "gene2go_Karl <- readMappings(\"from_karl_for_comzitNhua.GO_n_KEGG/hsal_NCBI_2018annot.b2g_annotations.for_topGO.v2.txt\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "971ea9c0",
   "metadata": {},
   "source": [
    "Import gene:GO mappings generated using CrowdGO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5085069e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# since parsed CrowdGO output does not have the topGO-compatible structure (see above),\n",
    "# import it using read.table() and then\n",
    "# create a list of vectors where each vector name is a gene name\n",
    "# and the vector contains all the GO-terms associated with the gene\n",
    "gene2go_CrowdGO_one_term_per_line <- read.table(\"/Users/bsierieb/Work-local/genomes/2021-04-20_HSAL51/CrowdGO/output_HSAL51/CrowdGO_bp/crowdgo_by_gene.tab\",\n",
    "                                                header=T)\n",
    "\n",
    "gene2go_CrowdGO <- list()\n",
    "for (i in 1:nrow(gene2go_CrowdGO_one_term_per_line)) {\n",
    "    gene <- gene2go_CrowdGO_one_term_per_line[i,1]\n",
    "    goterm <- gene2go_CrowdGO_one_term_per_line[i,2]\n",
    "    gene2go_CrowdGO[gene][[1]] <- c(gene2go_CrowdGO[gene][[1]], goterm)\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4ae3e4be",
   "metadata": {},
   "source": [
    "## Perform GO-term enrichment analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c5dfc33b",
   "metadata": {},
   "source": [
    "Define a function to sequentially perform BP GO enrichment analysis on up-, down-regulated, and all DE genes. This function is based off of a similar function written by Karl Glastad. Apart from renamed parameters and somewhat different code for some steps, no p-value adjustment is performed here. The reason is outlined in the [topGO manual](https://bioconductor.org/packages/devel/bioc/vignettes/topGO/inst/doc/topGO.pdf):\n",
    "\"For the methods that account for the GO topology **like elim** and weight, the problem of multiple\n",
    "testing is even more complicated. Here one computes the p-value of a GO term conditioned on the\n",
    "neighbouring terms. The tests are therefore not independent and the multiple testing theory does not\n",
    "directly apply. We like to interpret the p-values returned by these methods as corrected or not affected\n",
    "by multiple testing.\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "f1ad7cde",
   "metadata": {},
   "outputs": [],
   "source": [
    "# function that sets the cutoff for determining what is a DEG and what is not (padj < 0.01, here)\n",
    "topDiffGenes <- function(allScore){allScore<0.01}\n",
    "\n",
    "# function to perform enrichment analysis\n",
    "runTOPGO <- function(de_table, g2go_mapping, terms_to_plot=10){\n",
    "    \n",
    "    # loop through up- down-, and either- regulated genes\n",
    "    # and set the alternative (or none, in the case of \"BOTH\") p-value to 1\n",
    "    # so only up- or down-regulated genes are tested, respectively\n",
    "    for (DIREC in c(\"UP\",\"DOWN\",\"BOTH\")) {\n",
    "        \n",
    "        # extract a vector of p-values (replace alternative with 1 if applicable, see above)\n",
    "        # and assign gene names as element names in the vector\n",
    "        if (DIREC == \"DOWN\") {\n",
    "            geneList <- ifelse(test=de_table$logFC>0,\n",
    "                               yes=1,\n",
    "                               no=de_table$FDR) # set the p-value of all up-regulated genes to 1\n",
    "        } else if (DIREC == \"UP\") {\n",
    "            geneList <- ifelse(test=de_table$logFC<0,\n",
    "                               yes=1,\n",
    "                               no=de_table$FDR) # set the p-value of all down-regulated genes to 1\n",
    "        } else if (DIREC == \"BOTH\") {\n",
    "            geneList <- de_table$FDR\n",
    "        }\n",
    "        names(geneList) <- de_table$genes\n",
    "\n",
    "        # set up a GOdata object\n",
    "        GOdata <- new(\"topGOdata\",\n",
    "                      description=\"\",\n",
    "                      ontology=\"BP\",\n",
    "                      allGenes=geneList,\n",
    "                      geneSel=topDiffGenes,\n",
    "                      annot=annFUN.gene2GO,\n",
    "                      nodeSize=5,\n",
    "                      gene2GO=g2go_mapping)\n",
    "        print(GOdata) # for troubleshooting\n",
    "        \n",
    "        # run the elim version of Fisher's Exact Test:\n",
    "        # important so as to avoid extreme redundancy in GO-term annotations\n",
    "        resultFisher.elim <- runTest(GOdata,\n",
    "                                     algorithm=\"elim\",\n",
    "                                     statistic=\"Fisher\")\n",
    "        \n",
    "        # extract the number of significant (p < 0.01) p-values\n",
    "        n_significant_terms <- length(which(resultFisher.elim@score<0.01))\n",
    "        \n",
    "        # generate results table\n",
    "        res <- GenTable(GOdata,\n",
    "                        elimFisher=resultFisher.elim,\n",
    "                        topNodes=n_significant_terms,\n",
    "                        orderBy=\"elimFisher\",\n",
    "                        numChar=200)\n",
    "        \n",
    "        # convert the p-value column to numeric type\n",
    "        if (length(which(is.na(as.numeric(res$elimFisher))))>0) {\n",
    "            break # stop if there are any p-values that cannot be converted to numeric\n",
    "        } else {\n",
    "            res$elimFisher <- as.numeric(res$elimFisher)\n",
    "        }\n",
    "        \n",
    "        # write results to a file\n",
    "        filename <- paste0(\"topGO_output/\",\n",
    "                           deparse(substitute(de_table)),\n",
    "                           \"_\",\n",
    "                           deparse(substitute(g2go_mapping)),\n",
    "                           \"_\",\n",
    "                           DIREC,\n",
    "                           \".tsv\")\n",
    "        write.table(res,\n",
    "                    filename,\n",
    "                    sep=\"\\t\",\n",
    "                    row.names=F,\n",
    "                    quote=F)\n",
    "        \n",
    "        # plot -log(p-value) of top enriched GO terms\n",
    "        # select as many terms as specified in terms_to_plot,\n",
    "        # unless there are fewer significant terms\n",
    "        if (terms_to_plot<=n_significant_terms) {\n",
    "            plot_df <- res[1:terms_to_plot,]\n",
    "        } else {\n",
    "            plot_df <- res\n",
    "        }\n",
    "        \n",
    "        \n",
    "        plot_df$Term <- factor(plot_df$Term,\n",
    "                               levels=rev(plot_df$Term))\n",
    "        p1 <- ggplot(data=plot_df,\n",
    "                     mapping=aes(x=Term,\n",
    "                                 y=-log(elimFisher),\n",
    "                                 fill=-log(elimFisher))) +\n",
    "            geom_col() +\n",
    "            coord_flip() +\n",
    "            ggtitle(paste0(deparse(substitute(de_table)),\n",
    "                           \"_\",\n",
    "                           deparse(substitute(g2go_mapping)),\n",
    "                           capture.output(GOdata)[18])) # append the number of FEASIBLE significantly DE genes\n",
    "\n",
    "        filename <- paste0(\"topGO_output/\",\n",
    "                           deparse(substitute(de_table)),\n",
    "                           \"__\",\n",
    "                           deparse(substitute(g2go_mapping)),\n",
    "                           \"_\",\n",
    "                           DIREC,\n",
    "                           \".png\")\n",
    "        ggsave(filename,\n",
    "               p1,\n",
    "               width=15,\n",
    "               height=terms_to_plot/2)\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5e104c78",
   "metadata": {},
   "source": [
    "Perform enrichment analysis for different comparisons and with different gene-to-GO-term mappings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3f69270d",
   "metadata": {},
   "outputs": [],
   "source": [
    "dir.create(\"topGO_output\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "ef836c7a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105191040 LOC112590176 LOC109504628 LOC112590521 LOC105191801  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 4160  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105191040 LOC112590176 LOC109504628 LOC112590521 LOC105191801  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 3448  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 7440 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 19:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 18:\t4 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t8 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t26 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t70 nodes to be scored\t(8 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t146 nodes to be scored\t(154 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t217 nodes to be scored\t(499 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t379 nodes to be scored\t(1175 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t665 nodes to be scored\t(1482 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t962 nodes to be scored\t(2031 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t1123 nodes to be scored\t(2664 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1102 nodes to be scored\t(3450 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t1004 nodes to be scored\t(4226 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t820 nodes to be scored\t(5116 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t512 nodes to be scored\t(5749 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t261 nodes to be scored\t(5966 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t116 nodes to be scored\t(6207 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(6355 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(6363 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105191040 LOC112590176 LOC109504628 LOC112590521 LOC105191801  ...\n",
      "   - score :  2.78e-69 4.31e-60 1.91e-52 2.09e-51 1.31e-49  ...\n",
      "   - 149  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105191040 LOC112590176 LOC109504628 LOC112590521 LOC105191801  ...\n",
      "   - score :  2.78e-69 4.31e-60 1.91e-52 2.09e-51 1.31e-49  ...\n",
      "   - 94  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 2335 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 17:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t4 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t12 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t23 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t39 nodes to be scored\t(19 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t68 nodes to be scored\t(59 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t121 nodes to be scored\t(98 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t191 nodes to be scored\t(178 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t282 nodes to be scored\t(272 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t341 nodes to be scored\t(332 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t361 nodes to be scored\t(543 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t365 nodes to be scored\t(712 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t273 nodes to be scored\t(1663 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t156 nodes to be scored\t(1663 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t78 nodes to be scored\t(1716 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t19 nodes to be scored\t(1746 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(1746 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105191040 LOC112590176 LOC109504628 LOC112590521 LOC105191801  ...\n",
      "   - score :  2.78e-69 4.31e-60 1.91e-52 2.09e-51 1.31e-49  ...\n",
      "   - 4309  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105191040 LOC112590176 LOC109504628 LOC112590521 LOC105191801  ...\n",
      "   - score :  2.78e-69 4.31e-60 1.91e-52 2.09e-51 1.31e-49  ...\n",
      "   - 3542  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 7454 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 19:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 18:\t4 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t8 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t26 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t70 nodes to be scored\t(8 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t147 nodes to be scored\t(154 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t217 nodes to be scored\t(513 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t379 nodes to be scored\t(839 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t667 nodes to be scored\t(1188 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t965 nodes to be scored\t(1836 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t1124 nodes to be scored\t(2498 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1103 nodes to be scored\t(3408 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t1006 nodes to be scored\t(4272 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t820 nodes to be scored\t(5041 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t515 nodes to be scored\t(5764 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t262 nodes to be scored\t(6039 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t116 nodes to be scored\t(6231 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(6351 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(6355 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105191040 LOC112590176 LOC109504628 LOC112590521 LOC105191801  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 4160  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105191040 LOC112590176 LOC109504628 LOC112590521 LOC105191801  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 4072  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 6906 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t5 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t14 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t36 nodes to be scored\t(35 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t87 nodes to be scored\t(113 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t139 nodes to be scored\t(311 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t262 nodes to be scored\t(653 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t495 nodes to be scored\t(1141 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t817 nodes to be scored\t(1952 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t1027 nodes to be scored\t(2475 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1081 nodes to be scored\t(3162 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t1058 nodes to be scored\t(3965 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t907 nodes to be scored\t(4674 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t554 nodes to be scored\t(5229 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t275 nodes to be scored\t(5835 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t123 nodes to be scored\t(6766 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(6792 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(10849 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105191040 LOC112590176 LOC109504628 LOC112590521 LOC105191801  ...\n",
      "   - score :  2.78e-69 4.31e-60 1.91e-52 2.09e-51 1.31e-49  ...\n",
      "   - 149  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105191040 LOC112590176 LOC109504628 LOC112590521 LOC105191801  ...\n",
      "   - score :  2.78e-69 4.31e-60 1.91e-52 2.09e-51 1.31e-49  ...\n",
      "   - 145  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 1414 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 17:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t3 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t3 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t9 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t19 nodes to be scored\t(63 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t37 nodes to be scored\t(91 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t80 nodes to be scored\t(103 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t126 nodes to be scored\t(130 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t175 nodes to be scored\t(150 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t229 nodes to be scored\t(225 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t281 nodes to be scored\t(289 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t226 nodes to be scored\t(362 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t139 nodes to be scored\t(363 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t67 nodes to be scored\t(363 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t16 nodes to be scored\t(363 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(363 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105191040 LOC112590176 LOC109504628 LOC112590521 LOC105191801  ...\n",
      "   - score :  2.78e-69 4.31e-60 1.91e-52 2.09e-51 1.31e-49  ...\n",
      "   - 4309  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105191040 LOC112590176 LOC109504628 LOC112590521 LOC105191801  ...\n",
      "   - score :  2.78e-69 4.31e-60 1.91e-52 2.09e-51 1.31e-49  ...\n",
      "   - 4217  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 6906 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t5 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t14 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t36 nodes to be scored\t(35 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t87 nodes to be scored\t(102 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t139 nodes to be scored\t(304 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t262 nodes to be scored\t(722 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t495 nodes to be scored\t(1177 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t817 nodes to be scored\t(1908 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t1027 nodes to be scored\t(2429 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1081 nodes to be scored\t(3069 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t1058 nodes to be scored\t(3834 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t907 nodes to be scored\t(4652 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t554 nodes to be scored\t(5267 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t275 nodes to be scored\t(5714 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t123 nodes to be scored\t(6694 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(6719 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(10850 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC112588737 LOC112590521 LOC105186441 LOC105190430 LOC105190836  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 2425  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC112588737 LOC112590521 LOC105186441 LOC105190430 LOC105190836  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 1970  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 7111 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 19:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 18:\t4 nodes to be scored\t(14 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t8 nodes to be scored\t(30 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t25 nodes to be scored\t(30 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t65 nodes to be scored\t(84 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t135 nodes to be scored\t(253 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t194 nodes to be scored\t(535 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t344 nodes to be scored\t(1121 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t620 nodes to be scored\t(1768 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t908 nodes to be scored\t(2264 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t1078 nodes to be scored\t(2659 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1061 nodes to be scored\t(3253 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t981 nodes to be scored\t(3775 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t792 nodes to be scored\t(4290 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t498 nodes to be scored\t(5008 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t259 nodes to be scored\t(5599 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t114 nodes to be scored\t(5751 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(5927 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(5927 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC112588737 LOC112590521 LOC105186441 LOC105190430 LOC105190836  ...\n",
      "   - score :  1.72e-39 3.79e-38 5.43e-38 1.02e-35 2.3e-33  ...\n",
      "   - 113  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC112588737 LOC112590521 LOC105186441 LOC105190430 LOC105190836  ...\n",
      "   - score :  1.72e-39 3.79e-38 5.43e-38 1.02e-35 2.3e-33  ...\n",
      "   - 77  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 2058 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 17:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t11 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t19 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t29 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t48 nodes to be scored\t(77 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t90 nodes to be scored\t(100 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t158 nodes to be scored\t(207 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t243 nodes to be scored\t(248 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t295 nodes to be scored\t(348 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t328 nodes to be scored\t(622 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t341 nodes to be scored\t(769 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t253 nodes to be scored\t(1700 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t145 nodes to be scored\t(1700 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t74 nodes to be scored\t(2256 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t20 nodes to be scored\t(2256 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(2256 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC112588737 LOC112590521 LOC105186441 LOC105190430 LOC105190836  ...\n",
      "   - score :  1.72e-39 3.79e-38 5.43e-38 1.02e-35 2.3e-33  ...\n",
      "   - 2538  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC112588737 LOC112590521 LOC105186441 LOC105190430 LOC105190836  ...\n",
      "   - score :  1.72e-39 3.79e-38 5.43e-38 1.02e-35 2.3e-33  ...\n",
      "   - 2047  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 7136 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 19:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 18:\t4 nodes to be scored\t(14 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t8 nodes to be scored\t(30 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t25 nodes to be scored\t(30 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t65 nodes to be scored\t(60 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t136 nodes to be scored\t(247 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t194 nodes to be scored\t(371 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t345 nodes to be scored\t(832 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t623 nodes to be scored\t(1653 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t911 nodes to be scored\t(2233 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t1084 nodes to be scored\t(2609 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1065 nodes to be scored\t(3235 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t982 nodes to be scored\t(3793 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t794 nodes to be scored\t(4367 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t501 nodes to be scored\t(5374 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t259 nodes to be scored\t(5700 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t115 nodes to be scored\t(5816 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(5980 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(5981 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC112588737 LOC112590521 LOC105186441 LOC105190430 LOC105190836  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 2425  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC112588737 LOC112590521 LOC105186441 LOC105190430 LOC105190836  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 2387  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 6667 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t4 nodes to be scored\t(5 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t13 nodes to be scored\t(5 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t32 nodes to be scored\t(5 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t83 nodes to be scored\t(86 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t131 nodes to be scored\t(201 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t247 nodes to be scored\t(442 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t472 nodes to be scored\t(915 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t781 nodes to be scored\t(1526 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t988 nodes to be scored\t(1969 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1045 nodes to be scored\t(2505 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t1032 nodes to be scored\t(3108 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t886 nodes to be scored\t(3941 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t539 nodes to be scored\t(4583 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t266 nodes to be scored\t(5034 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t122 nodes to be scored\t(5271 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(5378 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(10708 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC112588737 LOC112590521 LOC105186441 LOC105190430 LOC105190836  ...\n",
      "   - score :  1.72e-39 3.79e-38 5.43e-38 1.02e-35 2.3e-33  ...\n",
      "   - 113  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC112588737 LOC112590521 LOC105186441 LOC105190430 LOC105190836  ...\n",
      "   - score :  1.72e-39 3.79e-38 5.43e-38 1.02e-35 2.3e-33  ...\n",
      "   - 111  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 1272 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 17:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t3 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t3 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t11 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t21 nodes to be scored\t(63 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t33 nodes to be scored\t(106 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t64 nodes to be scored\t(118 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t108 nodes to be scored\t(138 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t160 nodes to be scored\t(158 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t210 nodes to be scored\t(207 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t242 nodes to be scored\t(277 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t199 nodes to be scored\t(330 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t130 nodes to be scored\t(330 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t67 nodes to be scored\t(330 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t17 nodes to be scored\t(494 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(494 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC112588737 LOC112590521 LOC105186441 LOC105190430 LOC105190836  ...\n",
      "   - score :  1.72e-39 3.79e-38 5.43e-38 1.02e-35 2.3e-33  ...\n",
      "   - 2538  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC112588737 LOC112590521 LOC105186441 LOC105190430 LOC105190836  ...\n",
      "   - score :  1.72e-39 3.79e-38 5.43e-38 1.02e-35 2.3e-33  ...\n",
      "   - 2498  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 6677 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t4 nodes to be scored\t(5 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t13 nodes to be scored\t(5 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t32 nodes to be scored\t(5 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t83 nodes to be scored\t(86 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t131 nodes to be scored\t(193 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t247 nodes to be scored\t(491 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t472 nodes to be scored\t(733 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t781 nodes to be scored\t(1561 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t989 nodes to be scored\t(2042 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1049 nodes to be scored\t(2417 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t1035 nodes to be scored\t(3197 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t888 nodes to be scored\t(3962 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t539 nodes to be scored\t(4581 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t266 nodes to be scored\t(5042 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t122 nodes to be scored\t(5441 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(5607 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(10710 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105183918 LOC105189590 LOC105189997 LOC112588046 LOC105181759  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 2795  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105189590 LOC105181602 LOC105181150 LOC105180903 LOC105190303  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 2082  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 7119 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 19:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 18:\t4 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t8 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t25 nodes to be scored\t(11 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t67 nodes to be scored\t(66 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t138 nodes to be scored\t(172 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t200 nodes to be scored\t(551 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t358 nodes to be scored\t(1420 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t630 nodes to be scored\t(1725 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t914 nodes to be scored\t(2112 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t1058 nodes to be scored\t(2529 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1052 nodes to be scored\t(3085 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t972 nodes to be scored\t(3594 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t791 nodes to be scored\t(4355 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t506 nodes to be scored\t(4812 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t257 nodes to be scored\t(5139 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t114 nodes to be scored\t(5199 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(5281 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(5351 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105183918 LOC105189590 LOC105189997 LOC112588046 LOC105181759  ...\n",
      "   - score :  2.25e-97 2.49e-97 1.35e-89 1.67e-86 5.17e-83  ...\n",
      "   - 1307  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105189590 LOC105181602 LOC105181150 LOC105180903 LOC105190303  ...\n",
      "   - score :  2.49e-97 1.66e-80 1.33e-79 3.9e-79 1.58e-78  ...\n",
      "   - 927  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 5481 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t3 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t5 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t9 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t33 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t95 nodes to be scored\t(37 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t141 nodes to be scored\t(37 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t239 nodes to be scored\t(37 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t423 nodes to be scored\t(430 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t643 nodes to be scored\t(455 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t798 nodes to be scored\t(525 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t816 nodes to be scored\t(576 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t797 nodes to be scored\t(789 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t669 nodes to be scored\t(989 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t451 nodes to be scored\t(1792 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t231 nodes to be scored\t(2323 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t106 nodes to be scored\t(2363 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t21 nodes to be scored\t(2386 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(6195 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105183918 LOC105189590 LOC105189997 LOC112588046 LOC105181759  ...\n",
      "   - score :  2.25e-97 2.49e-97 1.35e-89 1.67e-86 5.17e-83  ...\n",
      "   - 4102  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105189590 LOC105181602 LOC105181150 LOC105180903 LOC105190303  ...\n",
      "   - score :  2.49e-97 1.66e-80 1.33e-79 3.9e-79 1.58e-78  ...\n",
      "   - 3009  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 7390 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 19:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 18:\t4 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t8 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t25 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t69 nodes to be scored\t(68 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t146 nodes to be scored\t(149 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t210 nodes to be scored\t(422 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t378 nodes to be scored\t(1285 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t661 nodes to be scored\t(1411 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t956 nodes to be scored\t(1831 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t1109 nodes to be scored\t(2276 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1095 nodes to be scored\t(2655 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t1003 nodes to be scored\t(3232 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t812 nodes to be scored\t(3854 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t513 nodes to be scored\t(4728 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t262 nodes to be scored\t(5044 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t114 nodes to be scored\t(5217 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(5667 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(5681 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105183918 LOC105189590 LOC105189997 LOC112588046 LOC105181759  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 2795  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105183918 LOC105189590 LOC105189997 LOC112588046 LOC105181759  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 2738  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 6680 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t5 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t14 nodes to be scored\t(33 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t34 nodes to be scored\t(33 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t81 nodes to be scored\t(70 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t126 nodes to be scored\t(200 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t245 nodes to be scored\t(352 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t470 nodes to be scored\t(597 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t789 nodes to be scored\t(1420 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t995 nodes to be scored\t(1877 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1051 nodes to be scored\t(2405 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t1024 nodes to be scored\t(2972 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t882 nodes to be scored\t(3505 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t548 nodes to be scored\t(3965 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t269 nodes to be scored\t(4336 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t121 nodes to be scored\t(4506 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(4536 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(4811 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105183918 LOC105189590 LOC105189997 LOC112588046 LOC105181759  ...\n",
      "   - score :  2.25e-97 2.49e-97 1.35e-89 1.67e-86 5.17e-83  ...\n",
      "   - 1307  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105183918 LOC105189590 LOC105189997 LOC112588046 LOC105181759  ...\n",
      "   - score :  2.25e-97 2.49e-97 1.35e-89 1.67e-86 5.17e-83  ...\n",
      "   - 1268  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 4365 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t5 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t12 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t27 nodes to be scored\t(7 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t70 nodes to be scored\t(7 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t135 nodes to be scored\t(49 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t247 nodes to be scored\t(170 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t440 nodes to be scored\t(194 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t585 nodes to be scored\t(206 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t681 nodes to be scored\t(248 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t705 nodes to be scored\t(268 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t664 nodes to be scored\t(310 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t435 nodes to be scored\t(363 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t227 nodes to be scored\t(366 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t107 nodes to be scored\t(366 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t21 nodes to be scored\t(366 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(10837 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105183918 LOC105189590 LOC105189997 LOC112588046 LOC105181759  ...\n",
      "   - score :  2.25e-97 2.49e-97 1.35e-89 1.67e-86 5.17e-83  ...\n",
      "   - 4102  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105183918 LOC105189590 LOC105189997 LOC112588046 LOC105181759  ...\n",
      "   - score :  2.25e-97 2.49e-97 1.35e-89 1.67e-86 5.17e-83  ...\n",
      "   - 4006  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 6842 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t5 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t14 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t36 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t83 nodes to be scored\t(72 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t135 nodes to be scored\t(136 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t265 nodes to be scored\t(365 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t488 nodes to be scored\t(603 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t811 nodes to be scored\t(1231 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t1014 nodes to be scored\t(1727 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1072 nodes to be scored\t(1992 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t1044 nodes to be scored\t(2722 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t900 nodes to be scored\t(3225 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t553 nodes to be scored\t(3607 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t273 nodes to be scored\t(4003 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t123 nodes to be scored\t(4280 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(4617 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(10705 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105181150 LOC112588788 LOC105190357 LOC112588046 LOC105181077  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 2480  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105181150 LOC112588788 LOC105190357 LOC105188829 LOC105181146  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 1844  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 6967 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 19:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 18:\t4 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t8 nodes to be scored\t(18 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t25 nodes to be scored\t(18 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t66 nodes to be scored\t(67 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t131 nodes to be scored\t(188 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t186 nodes to be scored\t(453 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t339 nodes to be scored\t(1375 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t607 nodes to be scored\t(1616 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t890 nodes to be scored\t(1924 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t1044 nodes to be scored\t(2338 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1026 nodes to be scored\t(2920 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t964 nodes to be scored\t(3472 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t781 nodes to be scored\t(4224 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t501 nodes to be scored\t(4824 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t256 nodes to be scored\t(5077 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t114 nodes to be scored\t(5179 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(5691 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(6264 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105181150 LOC112588788 LOC105190357 LOC112588046 LOC105181077  ...\n",
      "   - score :  1.03e-75 1.28e-71 4.11e-70 4.11e-70 2.35e-69  ...\n",
      "   - 1330  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105181150 LOC112588788 LOC105190357 LOC105188829 LOC105181146  ...\n",
      "   - score :  1.03e-75 1.28e-71 4.11e-70 4.43e-68 5.37e-67  ...\n",
      "   - 955  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 5628 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t4 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t9 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t34 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t100 nodes to be scored\t(37 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t152 nodes to be scored\t(58 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t258 nodes to be scored\t(122 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t452 nodes to be scored\t(427 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t680 nodes to be scored\t(607 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t820 nodes to be scored\t(778 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t825 nodes to be scored\t(820 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t802 nodes to be scored\t(1356 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t676 nodes to be scored\t(1413 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t454 nodes to be scored\t(2126 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t228 nodes to be scored\t(2326 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t110 nodes to be scored\t(2348 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t21 nodes to be scored\t(5962 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(5962 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105181150 LOC112588788 LOC105190357 LOC112588046 LOC105181077  ...\n",
      "   - score :  1.03e-75 1.28e-71 4.11e-70 4.11e-70 2.35e-69  ...\n",
      "   - 3810  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105181150 LOC112588788 LOC105190357 LOC105188829 LOC105181146  ...\n",
      "   - score :  1.03e-75 1.28e-71 4.11e-70 4.43e-68 5.37e-67  ...\n",
      "   - 2799  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 7348 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 19:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 18:\t4 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t8 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t25 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t69 nodes to be scored\t(37 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t147 nodes to be scored\t(94 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t208 nodes to be scored\t(180 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t374 nodes to be scored\t(1044 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t656 nodes to be scored\t(1438 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t952 nodes to be scored\t(1614 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t1109 nodes to be scored\t(1981 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1084 nodes to be scored\t(2333 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t997 nodes to be scored\t(2967 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t805 nodes to be scored\t(3568 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t508 nodes to be scored\t(4276 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t262 nodes to be scored\t(4919 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t115 nodes to be scored\t(5294 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(5342 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(5426 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105181150 LOC112588788 LOC105190357 LOC112588046 LOC105181077  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 2480  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105181150 LOC112588788 LOC105190357 LOC112588046 LOC105181077  ...\n",
      "   - score :  1 1 1 1 1  ...\n",
      "   - 2420  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 6592 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t5 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t14 nodes to be scored\t(18 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t34 nodes to be scored\t(29 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t79 nodes to be scored\t(57 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t123 nodes to be scored\t(181 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t240 nodes to be scored\t(268 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t457 nodes to be scored\t(561 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t771 nodes to be scored\t(1313 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t979 nodes to be scored\t(1783 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1041 nodes to be scored\t(2272 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t1018 nodes to be scored\t(2823 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t876 nodes to be scored\t(3419 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t541 nodes to be scored\t(3899 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t268 nodes to be scored\t(4255 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t120 nodes to be scored\t(4418 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(4492 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(4789 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105181150 LOC112588788 LOC105190357 LOC112588046 LOC105181077  ...\n",
      "   - score :  1.03e-75 1.28e-71 4.11e-70 4.11e-70 2.35e-69  ...\n",
      "   - 1330  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105181150 LOC112588788 LOC105190357 LOC112588046 LOC105181077  ...\n",
      "   - score :  1.03e-75 1.28e-71 4.11e-70 4.11e-70 2.35e-69  ...\n",
      "   - 1293  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 4549 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t5 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t11 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t30 nodes to be scored\t(7 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t81 nodes to be scored\t(7 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t145 nodes to be scored\t(49 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t262 nodes to be scored\t(109 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t460 nodes to be scored\t(109 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t628 nodes to be scored\t(169 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t703 nodes to be scored\t(174 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t733 nodes to be scored\t(185 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t691 nodes to be scored\t(210 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t444 nodes to be scored\t(226 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t227 nodes to be scored\t(228 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t105 nodes to be scored\t(323 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t21 nodes to be scored\t(323 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(10837 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105181150 LOC112588788 LOC105190357 LOC112588046 LOC105181077  ...\n",
      "   - score :  1.03e-75 1.28e-71 4.11e-70 4.11e-70 2.35e-69  ...\n",
      "   - 3810  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105181150 LOC112588788 LOC105190357 LOC112588046 LOC105181077  ...\n",
      "   - score :  1.03e-75 1.28e-71 4.11e-70 4.11e-70 2.35e-69  ...\n",
      "   - 3713  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 6812 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t5 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t14 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t36 nodes to be scored\t(35 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t84 nodes to be scored\t(35 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t135 nodes to be scored\t(107 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t263 nodes to be scored\t(167 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t490 nodes to be scored\t(522 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t809 nodes to be scored\t(730 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t1012 nodes to be scored\t(1180 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1063 nodes to be scored\t(1941 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t1037 nodes to be scored\t(2637 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t897 nodes to be scored\t(3212 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t548 nodes to be scored\t(3698 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t272 nodes to be scored\t(4080 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t121 nodes to be scored\t(4121 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(4274 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(10694 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105189919 LOC105185161 LOC109504166 LOC105184204 LOC105189488  ...\n",
      "   - score :  1 1 1 1 1.12e-15  ...\n",
      "   - 98  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105185161 LOC105189488 LOC105181726 LOC112589664 LOC105185443  ...\n",
      "   - score :  1 1.12e-15 1 7.87e-11 3.83e-10  ...\n",
      "   - 86  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 2385 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 19:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 18:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t3 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t7 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t10 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t26 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t51 nodes to be scored\t(39 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t82 nodes to be scored\t(50 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t119 nodes to be scored\t(65 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t187 nodes to be scored\t(373 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t281 nodes to be scored\t(1213 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t318 nodes to be scored\t(1354 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t367 nodes to be scored\t(1670 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t379 nodes to be scored\t(1841 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t291 nodes to be scored\t(2112 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t162 nodes to be scored\t(2844 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t78 nodes to be scored\t(2902 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t20 nodes to be scored\t(2908 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(2908 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105189919 LOC105185161 LOC109504166 LOC105184204 LOC105189488  ...\n",
      "   - score :  4.85e-22 4.85e-22 2.89e-17 8.14e-17 1  ...\n",
      "   - 42  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105185161 LOC105189488 LOC105181726 LOC112589664 LOC105185443  ...\n",
      "   - score :  4.85e-22 1 2.6e-12 1 1  ...\n",
      "   - 23  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 628 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 14:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t5 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t10 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t15 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t23 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t41 nodes to be scored\t(41 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t70 nodes to be scored\t(41 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t92 nodes to be scored\t(108 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t117 nodes to be scored\t(128 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t123 nodes to be scored\t(183 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t79 nodes to be scored\t(241 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t36 nodes to be scored\t(241 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t15 nodes to be scored\t(257 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(257 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105189919 LOC105185161 LOC109504166 LOC105184204 LOC105189488  ...\n",
      "   - score :  4.85e-22 4.85e-22 2.89e-17 8.14e-17 1.12e-15  ...\n",
      "   - 140  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105185161 LOC105189488 LOC105181726 LOC112589664 LOC105185443  ...\n",
      "   - score :  4.85e-22 1.12e-15 2.6e-12 7.87e-11 3.83e-10  ...\n",
      "   - 109  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 2533 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 19:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 18:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t3 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t7 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t10 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t26 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t52 nodes to be scored\t(11 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t86 nodes to be scored\t(233 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t126 nodes to be scored\t(244 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t199 nodes to be scored\t(272 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t302 nodes to be scored\t(574 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t352 nodes to be scored\t(840 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t395 nodes to be scored\t(967 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t401 nodes to be scored\t(1205 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t304 nodes to be scored\t(1496 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t168 nodes to be scored\t(2157 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t78 nodes to be scored\t(2273 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t20 nodes to be scored\t(2285 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(2285 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105189919 LOC105185161 LOC109504166 LOC105184204 LOC105189488  ...\n",
      "   - score :  1 1 1 1 1.12e-15  ...\n",
      "   - 98  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105189919 LOC105185161 LOC109504166 LOC105184204 LOC105189488  ...\n",
      "   - score :  1 1 1 1 1.12e-15  ...\n",
      "   - 96  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 2310 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t1 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t5 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t7 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t16 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t35 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t56 nodes to be scored\t(91 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t94 nodes to be scored\t(103 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t166 nodes to be scored\t(166 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t247 nodes to be scored\t(402 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t311 nodes to be scored\t(681 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t389 nodes to be scored\t(867 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t399 nodes to be scored\t(1072 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t310 nodes to be scored\t(2114 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t173 nodes to be scored\t(2143 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t79 nodes to be scored\t(3031 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t20 nodes to be scored\t(3210 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(3236 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105189919 LOC105185161 LOC109504166 LOC105184204 LOC105189488  ...\n",
      "   - score :  4.85e-22 4.85e-22 2.89e-17 8.14e-17 1  ...\n",
      "   - 42  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105189919 LOC105185161 LOC109504166 LOC105184204 LOC105189488  ...\n",
      "   - score :  4.85e-22 4.85e-22 2.89e-17 8.14e-17 1  ...\n",
      "   - 37  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 909 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 14:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t7 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t16 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t24 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t42 nodes to be scored\t(8 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t79 nodes to be scored\t(27 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t113 nodes to be scored\t(27 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t147 nodes to be scored\t(132 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t168 nodes to be scored\t(132 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t148 nodes to be scored\t(133 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t93 nodes to be scored\t(153 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t52 nodes to be scored\t(153 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t18 nodes to be scored\t(153 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(153 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105189919 LOC105185161 LOC109504166 LOC105184204 LOC105189488  ...\n",
      "   - score :  4.85e-22 4.85e-22 2.89e-17 8.14e-17 1.12e-15  ...\n",
      "   - 140  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105189919 LOC105185161 LOC109504166 LOC105184204 LOC105189488  ...\n",
      "   - score :  4.85e-22 4.85e-22 2.89e-17 8.14e-17 1.12e-15  ...\n",
      "   - 133  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 2533 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t1 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t5 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t7 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t17 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t39 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t64 nodes to be scored\t(91 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t106 nodes to be scored\t(125 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t186 nodes to be scored\t(176 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t281 nodes to be scored\t(375 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t344 nodes to be scored\t(529 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t429 nodes to be scored\t(663 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t434 nodes to be scored\t(806 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t327 nodes to be scored\t(1294 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t182 nodes to be scored\t(1613 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t89 nodes to be scored\t(1697 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t20 nodes to be scored\t(2551 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(2652 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105188415 LOC112589758 LOC105182798 LOC112589664 LOC105187009  ...\n",
      "   - score :  1 1 1.69e-34 3.69e-34 8.63e-34  ...\n",
      "   - 1400  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC112589758 LOC105182798 LOC112589664 LOC105187009 LOC105192616  ...\n",
      "   - score :  1 1.69e-34 3.69e-34 8.63e-34 7.23e-32  ...\n",
      "   - 1128  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 6393 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 19:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 18:\t4 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t8 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t25 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t58 nodes to be scored\t(22 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t118 nodes to be scored\t(162 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t171 nodes to be scored\t(553 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t296 nodes to be scored\t(1308 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t524 nodes to be scored\t(1485 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t779 nodes to be scored\t(1925 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t947 nodes to be scored\t(2266 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t950 nodes to be scored\t(2781 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t903 nodes to be scored\t(3497 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t746 nodes to be scored\t(4012 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t474 nodes to be scored\t(4631 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t256 nodes to be scored\t(5000 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t110 nodes to be scored\t(5367 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t22 nodes to be scored\t(5461 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(5461 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105188415 LOC112589758 LOC105182798 LOC112589664 LOC105187009  ...\n",
      "   - score :  8.15e-38 1.69e-34 1 1 1  ...\n",
      "   - 2407  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC112589758 LOC105182798 LOC112589664 LOC105187009 LOC105192616  ...\n",
      "   - score :  1.69e-34 1 1 1 1  ...\n",
      "   - 1886  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 6236 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t3 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t6 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t20 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t52 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t116 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t166 nodes to be scored\t(73 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t285 nodes to be scored\t(220 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t504 nodes to be scored\t(598 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t775 nodes to be scored\t(1043 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t924 nodes to be scored\t(1351 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t926 nodes to be scored\t(1750 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t891 nodes to be scored\t(2178 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t726 nodes to be scored\t(2312 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t466 nodes to be scored\t(3821 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t241 nodes to be scored\t(3877 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t112 nodes to be scored\t(3877 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t22 nodes to be scored\t(3913 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(3913 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 9924 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13627 GO terms and 30824 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 8833 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105188415 LOC112589758 LOC105182798 LOC112589664 LOC105187009  ...\n",
      "   - score :  8.15e-38 1.69e-34 1.69e-34 3.69e-34 8.63e-34  ...\n",
      "   - 3807  significant genes. \n",
      "\n",
      " 8833 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC112589758 LOC105182798 LOC112589664 LOC105187009 LOC105192616  ...\n",
      "   - score :  1.69e-34 1.69e-34 3.69e-34 8.63e-34 7.23e-32  ...\n",
      "   - 3014  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 7537 \n",
      "   - number of edges = 16510 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 7325 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 19:\t1 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 18:\t4 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t8 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t25 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t68 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t143 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t209 nodes to be scored\t(30 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t371 nodes to be scored\t(45 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t649 nodes to be scored\t(612 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t953 nodes to be scored\t(858 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t1104 nodes to be scored\t(1161 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1079 nodes to be scored\t(2025 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t992 nodes to be scored\t(2534 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t809 nodes to be scored\t(3735 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t510 nodes to be scored\t(4269 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t263 nodes to be scored\t(5185 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t114 nodes to be scored\t(5342 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t22 nodes to be scored\t(5370 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(5370 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105188415 LOC112589758 LOC105182798 LOC112589664 LOC105187009  ...\n",
      "   - score :  1 1 1.69e-34 3.69e-34 8.63e-34  ...\n",
      "   - 1400  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105188415 LOC105182798 LOC112589664 LOC105187009 LOC105192616  ...\n",
      "   - score :  1 1.69e-34 3.69e-34 8.63e-34 7.23e-32  ...\n",
      "   - 1379  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 5972 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t5 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t14 nodes to be scored\t(12 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t30 nodes to be scored\t(39 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t73 nodes to be scored\t(56 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t107 nodes to be scored\t(188 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t201 nodes to be scored\t(346 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t388 nodes to be scored\t(780 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t672 nodes to be scored\t(1293 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t869 nodes to be scored\t(1618 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t953 nodes to be scored\t(2094 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t936 nodes to be scored\t(2728 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t813 nodes to be scored\t(3612 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t511 nodes to be scored\t(4037 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t255 nodes to be scored\t(4329 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t119 nodes to be scored\t(4601 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(4857 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(5002 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105188415 LOC112589758 LOC105182798 LOC112589664 LOC105187009  ...\n",
      "   - score :  8.15e-38 1.69e-34 1 1 1  ...\n",
      "   - 2407  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105188415 LOC105182798 LOC112589664 LOC105187009 LOC105192616  ...\n",
      "   - score :  8.15e-38 1 1 1 1  ...\n",
      "   - 2354  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 5918 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t5 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t10 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t24 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t65 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t121 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t220 nodes to be scored\t(111 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t399 nodes to be scored\t(288 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t682 nodes to be scored\t(595 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t864 nodes to be scored\t(878 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t913 nodes to be scored\t(1213 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t919 nodes to be scored\t(1518 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t801 nodes to be scored\t(2625 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t495 nodes to be scored\t(2789 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t259 nodes to be scored\t(3420 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t116 nodes to be scored\t(4824 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t22 nodes to be scored\t(4838 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(10837 eliminated genes)\n",
      "\n",
      "\n",
      "Building most specific GOs .....\n",
      "\n",
      "\t( 8245 GO terms found. )\n",
      "\n",
      "\n",
      "Build GO DAG topology ..........\n",
      "\n",
      "\t( 13360 GO terms and 30383 relations. )\n",
      "\n",
      "\n",
      "Annotating nodes ...............\n",
      "\n",
      "\t( 12100 genes annotated to the GO terms. )\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n",
      " Description:\n",
      "   -   \n",
      "\n",
      " Ontology:\n",
      "   -  BP \n",
      "\n",
      " 12654 available genes (all genes from the array):\n",
      "   - symbol:  LOC105188415 LOC112589758 LOC105182798 LOC112589664 LOC105187009  ...\n",
      "   - score :  8.15e-38 1.69e-34 1.69e-34 3.69e-34 8.63e-34  ...\n",
      "   - 3807  significant genes. \n",
      "\n",
      " 12100 feasible genes (genes that can be used in the analysis):\n",
      "   - symbol:  LOC105188415 LOC105182798 LOC112589664 LOC105187009 LOC105192616  ...\n",
      "   - score :  8.15e-38 1.69e-34 3.69e-34 8.63e-34 7.23e-32  ...\n",
      "   - 3733  significant genes. \n",
      "\n",
      " GO graph (nodes with at least  5  genes):\n",
      "   - a graph with directed edges\n",
      "   - number of nodes = 6974 \n",
      "   - number of edges = 15439 \n",
      "\n",
      "------------------------- topGOdata object -------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\t\t\t -- Elim Algorithm -- \n",
      "\n",
      "\t\t the algorithm is scoring 6837 nontrivial nodes\n",
      "\t\t parameters: \n",
      "\t\t\t test statistic: fisher\n",
      "\t\t\t cutOff: 0.01\n",
      "\n",
      "\n",
      "\t Level 18:\t2 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 17:\t5 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 16:\t14 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 15:\t35 nodes to be scored\t(0 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 14:\t87 nodes to be scored\t(18 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 13:\t142 nodes to be scored\t(55 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 12:\t264 nodes to be scored\t(229 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 11:\t491 nodes to be scored\t(307 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 10:\t812 nodes to be scored\t(1126 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 9:\t1017 nodes to be scored\t(1521 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 8:\t1066 nodes to be scored\t(2416 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 7:\t1041 nodes to be scored\t(2936 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 6:\t895 nodes to be scored\t(3653 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 5:\t549 nodes to be scored\t(4000 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 4:\t270 nodes to be scored\t(5174 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 3:\t123 nodes to be scored\t(6160 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 2:\t23 nodes to be scored\t(6392 eliminated genes)\n",
      "\n",
      "\n",
      "\t Level 1:\t1 nodes to be scored\t(10848 eliminated genes)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "runTOPGO(de_table=Fb_W_vs_G,\n",
    "         g2go_mapping=gene2go_Karl,\n",
    "         terms_to_plot=20)\n",
    "runTOPGO(Fb_W_vs_G, gene2go_CrowdGO, 20)\n",
    "runTOPGO(Fb_G_vs_R, gene2go_Karl, 20)\n",
    "runTOPGO(Fb_G_vs_R, gene2go_CrowdGO, 20)\n",
    "runTOPGO(Ov_W_vs_G, gene2go_Karl, 20)\n",
    "runTOPGO(Ov_W_vs_G, gene2go_CrowdGO, 20)\n",
    "runTOPGO(Ov_G_vs_R, gene2go_Karl, 20)\n",
    "runTOPGO(Ov_G_vs_R, gene2go_CrowdGO, 20)\n",
    "runTOPGO(cBr_W_vs_G, gene2go_Karl, 20)\n",
    "runTOPGO(cBr_W_vs_G, gene2go_CrowdGO, 20)\n",
    "runTOPGO(cBr_G_vs_R, gene2go_Karl, 20)\n",
    "runTOPGO(cBr_G_vs_R, gene2go_CrowdGO, 20)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9db5cecd",
   "metadata": {},
   "source": [
    "Manually combine all .tsv files produced with CrowdGO mappings into an Excel file `__CrowdGO_all_stats.xlsx`"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "f354c0f7",
   "metadata": {},
   "source": [
    "One of the GO-terms, GO:0046949 (fatty-acyl-CoA biosynthetic process), attracts particular attention because it is the most enriched GO-term among the genes up-regulated in the gamergate fat body (i.e. down-regulated in both workers and revertants). Identify which DE genes drive the enrichment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 354,
   "id": "9d48c2fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "62"
      ],
      "text/latex": [
       "62"
      ],
      "text/markdown": [
       "62"
      ],
      "text/plain": [
       "[1] 62"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'LOC105180902'</li><li>'LOC105180907'</li><li>'LOC105180979'</li><li>'LOC105183157'</li><li>'LOC105183580'</li><li>'LOC105183580.3.p3'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'LOC105180902'\n",
       "\\item 'LOC105180907'\n",
       "\\item 'LOC105180979'\n",
       "\\item 'LOC105183157'\n",
       "\\item 'LOC105183580'\n",
       "\\item 'LOC105183580.3.p3'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'LOC105180902'\n",
       "2. 'LOC105180907'\n",
       "3. 'LOC105180979'\n",
       "4. 'LOC105183157'\n",
       "5. 'LOC105183580'\n",
       "6. 'LOC105183580.3.p3'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"LOC105180902\"      \"LOC105180907\"      \"LOC105180979\"     \n",
       "[4] \"LOC105183157\"      \"LOC105183580\"      \"LOC105183580.3.p3\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# find all genes in gene2go_CrowdGO associated with the GO-term\n",
    "FA_row_indices <- gene2go_CrowdGO_one_term_per_line$GO.term=='GO:0046949'\n",
    "FA_genes <- gene2go_CrowdGO_one_term_per_line$Protein[FA_row_indices]\n",
    "length(FA_genes) # check\n",
    "head(FA_genes) # check"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "02dfc8d4",
   "metadata": {},
   "source": [
    "Note that `FA_genes` contains both \"LOC105183580\" and \"LOC105183580.3.p3\". The latter is actually an isoform of the former. The reason for this is the following: CrowdGO predictions were initially generated for each ***transcript*** in `/Users/bsierieb/Work-local/genomes/2021-04-20_HSAL51/HSAL51.transcripts.fasta.transdecoder.pep`. Then, GO-terms were summarized by ***gene*** using a gene-to-transcript lookup table made from the annotation file `/Users/bsierieb/Work-local/genomes/2021-04-20_HSAL51/HSAL51.final.gtf`. The .gtf omits some transcripts contained in the .transdecoder.pep file, e.g. LOC105183580.3.p (I assume that Emily Shields did some additional cleanup when making the .gtf). Therefore, the CrowdGO annotations generally contain gene-to-GO mappings but also some transcript-to-GO mappings for the transcripts that could not be matched to any gene. I believe this has no consequences for the enrichment analysis as transcript-to-GO mappings will be simply ignored.\n",
    "\n",
    "See also:\n",
    "`/Users/bsierieb/Work-local/genomes/2021-04-20_HSAL51/CrowdGO/summarize_by_gene.R`\n",
    "`/Users/bsierieb/Work-local/genomes/2021-04-20_HSAL51/HSAL51.final_gene2tx.sh`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 360,
   "id": "db06ab88",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 13 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genes</th><th scope=col>logFC</th><th scope=col>logCPM</th><th scope=col>PValue</th><th scope=col>FDR</th><th scope=col>Annotation</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>LOC105191040</td><td>-9.456765</td><td> 8.6288891</td><td>2.190000e-73</td><td>2.780000e-69</td><td>fatty acid synthase                                              </td></tr>\n",
       "\t<tr><td>LOC105191801</td><td>-8.094335</td><td> 9.1392166</td><td>5.170000e-53</td><td>1.310000e-49</td><td>acyl-CoA Delta(11) desaturase                                    </td></tr>\n",
       "\t<tr><td>LOC105190430</td><td>-8.367141</td><td> 6.5409884</td><td>4.810000e-51</td><td>8.690000e-48</td><td>acyl-CoA Delta(11) desaturase                                    </td></tr>\n",
       "\t<tr><td>LOC105182047</td><td>-7.854964</td><td> 2.9948256</td><td>3.410000e-23</td><td>1.720000e-20</td><td>elongation of very long chain fatty acids protein AAEL008004     </td></tr>\n",
       "\t<tr><td>LOC105182544</td><td>-5.977819</td><td> 8.2981666</td><td>5.840000e-16</td><td>1.230000e-13</td><td>fatty acid synthase                                              </td></tr>\n",
       "\t<tr><td>LOC105185296</td><td>-4.198066</td><td> 1.1288395</td><td>8.870000e-10</td><td>3.880000e-08</td><td>elongation of very long chain fatty acids protein 7              </td></tr>\n",
       "\t<tr><td>LOC112590320</td><td>-4.002081</td><td> 7.1111678</td><td>4.160000e-09</td><td>1.470000e-07</td><td>fatty acid synthase-like isoform X1                              </td></tr>\n",
       "\t<tr><td>LOC105180907</td><td>-3.996957</td><td> 6.0816650</td><td>5.510000e-09</td><td>1.890000e-07</td><td>acyl-CoA Delta(11) desaturase isoform X2                         </td></tr>\n",
       "\t<tr><td>LOC105185168</td><td>-1.964630</td><td> 9.5122145</td><td>7.840000e-05</td><td>5.341220e-04</td><td>fatty acid synthase                                              </td></tr>\n",
       "\t<tr><td>LOC112590305</td><td>-4.139110</td><td> 0.5728714</td><td>1.597420e-04</td><td>9.388660e-04</td><td>fatty acid synthase-like                                         </td></tr>\n",
       "\t<tr><td>LOC105186441</td><td>-2.846044</td><td> 7.2347044</td><td>3.309670e-04</td><td>1.637240e-03</td><td>fatty acid synthase-like                                         </td></tr>\n",
       "\t<tr><td>LOC105180672</td><td>-2.986761</td><td> 9.5887926</td><td>6.896250e-04</td><td>2.880991e-03</td><td>fatty acid synthase                                              </td></tr>\n",
       "\t<tr><td>LOC112590548</td><td>-3.712446</td><td>-0.2990144</td><td>2.010479e-03</td><td>6.673819e-03</td><td>elongation of very long chain fatty acids protein AAEL008004-like</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 13 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " genes & logFC & logCPM & PValue & FDR & Annotation\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t LOC105191040 & -9.456765 &  8.6288891 & 2.190000e-73 & 2.780000e-69 & fatty acid synthase                                              \\\\\n",
       "\t LOC105191801 & -8.094335 &  9.1392166 & 5.170000e-53 & 1.310000e-49 & acyl-CoA Delta(11) desaturase                                    \\\\\n",
       "\t LOC105190430 & -8.367141 &  6.5409884 & 4.810000e-51 & 8.690000e-48 & acyl-CoA Delta(11) desaturase                                    \\\\\n",
       "\t LOC105182047 & -7.854964 &  2.9948256 & 3.410000e-23 & 1.720000e-20 & elongation of very long chain fatty acids protein AAEL008004     \\\\\n",
       "\t LOC105182544 & -5.977819 &  8.2981666 & 5.840000e-16 & 1.230000e-13 & fatty acid synthase                                              \\\\\n",
       "\t LOC105185296 & -4.198066 &  1.1288395 & 8.870000e-10 & 3.880000e-08 & elongation of very long chain fatty acids protein 7              \\\\\n",
       "\t LOC112590320 & -4.002081 &  7.1111678 & 4.160000e-09 & 1.470000e-07 & fatty acid synthase-like isoform X1                              \\\\\n",
       "\t LOC105180907 & -3.996957 &  6.0816650 & 5.510000e-09 & 1.890000e-07 & acyl-CoA Delta(11) desaturase isoform X2                         \\\\\n",
       "\t LOC105185168 & -1.964630 &  9.5122145 & 7.840000e-05 & 5.341220e-04 & fatty acid synthase                                              \\\\\n",
       "\t LOC112590305 & -4.139110 &  0.5728714 & 1.597420e-04 & 9.388660e-04 & fatty acid synthase-like                                         \\\\\n",
       "\t LOC105186441 & -2.846044 &  7.2347044 & 3.309670e-04 & 1.637240e-03 & fatty acid synthase-like                                         \\\\\n",
       "\t LOC105180672 & -2.986761 &  9.5887926 & 6.896250e-04 & 2.880991e-03 & fatty acid synthase                                              \\\\\n",
       "\t LOC112590548 & -3.712446 & -0.2990144 & 2.010479e-03 & 6.673819e-03 & elongation of very long chain fatty acids protein AAEL008004-like\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 13 × 6\n",
       "\n",
       "| genes &lt;chr&gt; | logFC &lt;dbl&gt; | logCPM &lt;dbl&gt; | PValue &lt;dbl&gt; | FDR &lt;dbl&gt; | Annotation &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| LOC105191040 | -9.456765 |  8.6288891 | 2.190000e-73 | 2.780000e-69 | fatty acid synthase                                               |\n",
       "| LOC105191801 | -8.094335 |  9.1392166 | 5.170000e-53 | 1.310000e-49 | acyl-CoA Delta(11) desaturase                                     |\n",
       "| LOC105190430 | -8.367141 |  6.5409884 | 4.810000e-51 | 8.690000e-48 | acyl-CoA Delta(11) desaturase                                     |\n",
       "| LOC105182047 | -7.854964 |  2.9948256 | 3.410000e-23 | 1.720000e-20 | elongation of very long chain fatty acids protein AAEL008004      |\n",
       "| LOC105182544 | -5.977819 |  8.2981666 | 5.840000e-16 | 1.230000e-13 | fatty acid synthase                                               |\n",
       "| LOC105185296 | -4.198066 |  1.1288395 | 8.870000e-10 | 3.880000e-08 | elongation of very long chain fatty acids protein 7               |\n",
       "| LOC112590320 | -4.002081 |  7.1111678 | 4.160000e-09 | 1.470000e-07 | fatty acid synthase-like isoform X1                               |\n",
       "| LOC105180907 | -3.996957 |  6.0816650 | 5.510000e-09 | 1.890000e-07 | acyl-CoA Delta(11) desaturase isoform X2                          |\n",
       "| LOC105185168 | -1.964630 |  9.5122145 | 7.840000e-05 | 5.341220e-04 | fatty acid synthase                                               |\n",
       "| LOC112590305 | -4.139110 |  0.5728714 | 1.597420e-04 | 9.388660e-04 | fatty acid synthase-like                                          |\n",
       "| LOC105186441 | -2.846044 |  7.2347044 | 3.309670e-04 | 1.637240e-03 | fatty acid synthase-like                                          |\n",
       "| LOC105180672 | -2.986761 |  9.5887926 | 6.896250e-04 | 2.880991e-03 | fatty acid synthase                                               |\n",
       "| LOC112590548 | -3.712446 | -0.2990144 | 2.010479e-03 | 6.673819e-03 | elongation of very long chain fatty acids protein AAEL008004-like |\n",
       "\n"
      ],
      "text/plain": [
       "   genes        logFC     logCPM     PValue       FDR         \n",
       "1  LOC105191040 -9.456765  8.6288891 2.190000e-73 2.780000e-69\n",
       "2  LOC105191801 -8.094335  9.1392166 5.170000e-53 1.310000e-49\n",
       "3  LOC105190430 -8.367141  6.5409884 4.810000e-51 8.690000e-48\n",
       "4  LOC105182047 -7.854964  2.9948256 3.410000e-23 1.720000e-20\n",
       "5  LOC105182544 -5.977819  8.2981666 5.840000e-16 1.230000e-13\n",
       "6  LOC105185296 -4.198066  1.1288395 8.870000e-10 3.880000e-08\n",
       "7  LOC112590320 -4.002081  7.1111678 4.160000e-09 1.470000e-07\n",
       "8  LOC105180907 -3.996957  6.0816650 5.510000e-09 1.890000e-07\n",
       "9  LOC105185168 -1.964630  9.5122145 7.840000e-05 5.341220e-04\n",
       "10 LOC112590305 -4.139110  0.5728714 1.597420e-04 9.388660e-04\n",
       "11 LOC105186441 -2.846044  7.2347044 3.309670e-04 1.637240e-03\n",
       "12 LOC105180672 -2.986761  9.5887926 6.896250e-04 2.880991e-03\n",
       "13 LOC112590548 -3.712446 -0.2990144 2.010479e-03 6.673819e-03\n",
       "   Annotation                                                       \n",
       "1  fatty acid synthase                                              \n",
       "2  acyl-CoA Delta(11) desaturase                                    \n",
       "3  acyl-CoA Delta(11) desaturase                                    \n",
       "4  elongation of very long chain fatty acids protein AAEL008004     \n",
       "5  fatty acid synthase                                              \n",
       "6  elongation of very long chain fatty acids protein 7              \n",
       "7  fatty acid synthase-like isoform X1                              \n",
       "8  acyl-CoA Delta(11) desaturase isoform X2                         \n",
       "9  fatty acid synthase                                              \n",
       "10 fatty acid synthase-like                                         \n",
       "11 fatty acid synthase-like                                         \n",
       "12 fatty acid synthase                                              \n",
       "13 elongation of very long chain fatty acids protein AAEL008004-like"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 13 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genes</th><th scope=col>logFC</th><th scope=col>logCPM</th><th scope=col>PValue</th><th scope=col>FDR</th><th scope=col>Annotation</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>LOC112588737</td><td>-9.051802</td><td>7.0899687</td><td>1.36000e-43</td><td>1.720000e-39</td><td>fatty acid synthase-like                                    </td></tr>\n",
       "\t<tr><td>LOC105186441</td><td>-8.216428</td><td>7.0494603</td><td>1.29000e-41</td><td>5.430000e-38</td><td>fatty acid synthase-like                                    </td></tr>\n",
       "\t<tr><td>LOC105190430</td><td>-8.864538</td><td>6.5357242</td><td>3.21000e-39</td><td>1.020000e-35</td><td>acyl-CoA Delta(11) desaturase                               </td></tr>\n",
       "\t<tr><td>LOC105191801</td><td>-7.277788</td><td>9.1423545</td><td>1.14000e-36</td><td>2.400000e-33</td><td>acyl-CoA Delta(11) desaturase                               </td></tr>\n",
       "\t<tr><td>LOC105180672</td><td>-7.382810</td><td>9.4255614</td><td>2.33000e-21</td><td>2.950000e-18</td><td>fatty acid synthase                                         </td></tr>\n",
       "\t<tr><td>LOC105182047</td><td>-6.689554</td><td>2.9650885</td><td>1.98000e-19</td><td>1.930000e-16</td><td>elongation of very long chain fatty acids protein AAEL008004</td></tr>\n",
       "\t<tr><td>LOC105180907</td><td>-4.608542</td><td>6.0474526</td><td>1.01000e-09</td><td>1.030000e-07</td><td>acyl-CoA Delta(11) desaturase isoform X2                    </td></tr>\n",
       "\t<tr><td>LOC112590175</td><td>-5.271302</td><td>5.6194296</td><td>9.93000e-09</td><td>7.090000e-07</td><td>fatty acid synthase-like                                    </td></tr>\n",
       "\t<tr><td>LOC105183865</td><td>-3.836275</td><td>5.2417284</td><td>1.31000e-08</td><td>8.720000e-07</td><td>fatty acid synthase-like isoform X1                         </td></tr>\n",
       "\t<tr><td>LOC112590320</td><td>-4.272346</td><td>7.0937949</td><td>8.17000e-08</td><td>3.840000e-06</td><td>fatty acid synthase-like isoform X1                         </td></tr>\n",
       "\t<tr><td>LOC105185296</td><td>-3.813467</td><td>1.0640692</td><td>3.72000e-07</td><td>1.340000e-05</td><td>elongation of very long chain fatty acids protein 7         </td></tr>\n",
       "\t<tr><td>LOC112590305</td><td>-3.561177</td><td>0.4730043</td><td>1.16370e-04</td><td>1.261819e-03</td><td>fatty acid synthase-like                                    </td></tr>\n",
       "\t<tr><td>LOC105191892</td><td>-2.843582</td><td>8.0037357</td><td>1.52678e-04</td><td>1.527263e-03</td><td>LOW QUALITY PROTEIN: fatty acid synthase-like               </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 13 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " genes & logFC & logCPM & PValue & FDR & Annotation\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t LOC112588737 & -9.051802 & 7.0899687 & 1.36000e-43 & 1.720000e-39 & fatty acid synthase-like                                    \\\\\n",
       "\t LOC105186441 & -8.216428 & 7.0494603 & 1.29000e-41 & 5.430000e-38 & fatty acid synthase-like                                    \\\\\n",
       "\t LOC105190430 & -8.864538 & 6.5357242 & 3.21000e-39 & 1.020000e-35 & acyl-CoA Delta(11) desaturase                               \\\\\n",
       "\t LOC105191801 & -7.277788 & 9.1423545 & 1.14000e-36 & 2.400000e-33 & acyl-CoA Delta(11) desaturase                               \\\\\n",
       "\t LOC105180672 & -7.382810 & 9.4255614 & 2.33000e-21 & 2.950000e-18 & fatty acid synthase                                         \\\\\n",
       "\t LOC105182047 & -6.689554 & 2.9650885 & 1.98000e-19 & 1.930000e-16 & elongation of very long chain fatty acids protein AAEL008004\\\\\n",
       "\t LOC105180907 & -4.608542 & 6.0474526 & 1.01000e-09 & 1.030000e-07 & acyl-CoA Delta(11) desaturase isoform X2                    \\\\\n",
       "\t LOC112590175 & -5.271302 & 5.6194296 & 9.93000e-09 & 7.090000e-07 & fatty acid synthase-like                                    \\\\\n",
       "\t LOC105183865 & -3.836275 & 5.2417284 & 1.31000e-08 & 8.720000e-07 & fatty acid synthase-like isoform X1                         \\\\\n",
       "\t LOC112590320 & -4.272346 & 7.0937949 & 8.17000e-08 & 3.840000e-06 & fatty acid synthase-like isoform X1                         \\\\\n",
       "\t LOC105185296 & -3.813467 & 1.0640692 & 3.72000e-07 & 1.340000e-05 & elongation of very long chain fatty acids protein 7         \\\\\n",
       "\t LOC112590305 & -3.561177 & 0.4730043 & 1.16370e-04 & 1.261819e-03 & fatty acid synthase-like                                    \\\\\n",
       "\t LOC105191892 & -2.843582 & 8.0037357 & 1.52678e-04 & 1.527263e-03 & LOW QUALITY PROTEIN: fatty acid synthase-like               \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 13 × 6\n",
       "\n",
       "| genes &lt;chr&gt; | logFC &lt;dbl&gt; | logCPM &lt;dbl&gt; | PValue &lt;dbl&gt; | FDR &lt;dbl&gt; | Annotation &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| LOC112588737 | -9.051802 | 7.0899687 | 1.36000e-43 | 1.720000e-39 | fatty acid synthase-like                                     |\n",
       "| LOC105186441 | -8.216428 | 7.0494603 | 1.29000e-41 | 5.430000e-38 | fatty acid synthase-like                                     |\n",
       "| LOC105190430 | -8.864538 | 6.5357242 | 3.21000e-39 | 1.020000e-35 | acyl-CoA Delta(11) desaturase                                |\n",
       "| LOC105191801 | -7.277788 | 9.1423545 | 1.14000e-36 | 2.400000e-33 | acyl-CoA Delta(11) desaturase                                |\n",
       "| LOC105180672 | -7.382810 | 9.4255614 | 2.33000e-21 | 2.950000e-18 | fatty acid synthase                                          |\n",
       "| LOC105182047 | -6.689554 | 2.9650885 | 1.98000e-19 | 1.930000e-16 | elongation of very long chain fatty acids protein AAEL008004 |\n",
       "| LOC105180907 | -4.608542 | 6.0474526 | 1.01000e-09 | 1.030000e-07 | acyl-CoA Delta(11) desaturase isoform X2                     |\n",
       "| LOC112590175 | -5.271302 | 5.6194296 | 9.93000e-09 | 7.090000e-07 | fatty acid synthase-like                                     |\n",
       "| LOC105183865 | -3.836275 | 5.2417284 | 1.31000e-08 | 8.720000e-07 | fatty acid synthase-like isoform X1                          |\n",
       "| LOC112590320 | -4.272346 | 7.0937949 | 8.17000e-08 | 3.840000e-06 | fatty acid synthase-like isoform X1                          |\n",
       "| LOC105185296 | -3.813467 | 1.0640692 | 3.72000e-07 | 1.340000e-05 | elongation of very long chain fatty acids protein 7          |\n",
       "| LOC112590305 | -3.561177 | 0.4730043 | 1.16370e-04 | 1.261819e-03 | fatty acid synthase-like                                     |\n",
       "| LOC105191892 | -2.843582 | 8.0037357 | 1.52678e-04 | 1.527263e-03 | LOW QUALITY PROTEIN: fatty acid synthase-like                |\n",
       "\n"
      ],
      "text/plain": [
       "   genes        logFC     logCPM    PValue      FDR         \n",
       "1  LOC112588737 -9.051802 7.0899687 1.36000e-43 1.720000e-39\n",
       "2  LOC105186441 -8.216428 7.0494603 1.29000e-41 5.430000e-38\n",
       "3  LOC105190430 -8.864538 6.5357242 3.21000e-39 1.020000e-35\n",
       "4  LOC105191801 -7.277788 9.1423545 1.14000e-36 2.400000e-33\n",
       "5  LOC105180672 -7.382810 9.4255614 2.33000e-21 2.950000e-18\n",
       "6  LOC105182047 -6.689554 2.9650885 1.98000e-19 1.930000e-16\n",
       "7  LOC105180907 -4.608542 6.0474526 1.01000e-09 1.030000e-07\n",
       "8  LOC112590175 -5.271302 5.6194296 9.93000e-09 7.090000e-07\n",
       "9  LOC105183865 -3.836275 5.2417284 1.31000e-08 8.720000e-07\n",
       "10 LOC112590320 -4.272346 7.0937949 8.17000e-08 3.840000e-06\n",
       "11 LOC105185296 -3.813467 1.0640692 3.72000e-07 1.340000e-05\n",
       "12 LOC112590305 -3.561177 0.4730043 1.16370e-04 1.261819e-03\n",
       "13 LOC105191892 -2.843582 8.0037357 1.52678e-04 1.527263e-03\n",
       "   Annotation                                                  \n",
       "1  fatty acid synthase-like                                    \n",
       "2  fatty acid synthase-like                                    \n",
       "3  acyl-CoA Delta(11) desaturase                               \n",
       "4  acyl-CoA Delta(11) desaturase                               \n",
       "5  fatty acid synthase                                         \n",
       "6  elongation of very long chain fatty acids protein AAEL008004\n",
       "7  acyl-CoA Delta(11) desaturase isoform X2                    \n",
       "8  fatty acid synthase-like                                    \n",
       "9  fatty acid synthase-like isoform X1                         \n",
       "10 fatty acid synthase-like isoform X1                         \n",
       "11 elongation of very long chain fatty acids protein 7         \n",
       "12 fatty acid synthase-like                                    \n",
       "13 LOW QUALITY PROTEIN: fatty acid synthase-like               "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# find which of these genes are significantly down-regulated in the W vs G comparison\n",
    "Fb_W_vs_G_FA_genes <- Fb_W_vs_G %>%\n",
    " filter(FDR<0.01) %>%\n",
    " filter(logFC<0) %>%\n",
    " filter(genes %in% FA_genes)\n",
    "Fb_W_vs_G_FA_genes # check\n",
    "\n",
    "# find which of these genes are significantly down-regulated in the R vs G comparison\n",
    "Fb_G_vs_R_FA_genes <- Fb_G_vs_R %>%\n",
    " filter(FDR<0.01) %>%\n",
    " filter(logFC<0) %>%\n",
    " filter(genes %in% FA_genes)\n",
    "Fb_G_vs_R_FA_genes # check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 365,
   "id": "eaf678c4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 17 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genes</th><th scope=col>logFC_W_vs_G</th><th scope=col>logCPM_W_vs_G</th><th scope=col>PValue_W_vs_G</th><th scope=col>FDR_W_vs_G</th><th scope=col>Annotation</th><th scope=col>logFC_R_vs_G</th><th scope=col>logCPM_R_vs_G</th><th scope=col>PValue_R_vs_G</th><th scope=col>FDR_R_vs_G</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>LOC105191040</td><td>-9.456765</td><td> 8.6288891</td><td>2.190000e-73</td><td>2.780000e-69</td><td>fatty acid synthase                                              </td><td>       NA</td><td>       NA</td><td>         NA</td><td>          NA</td></tr>\n",
       "\t<tr><td>LOC105191801</td><td>-8.094335</td><td> 9.1392166</td><td>5.170000e-53</td><td>1.310000e-49</td><td>acyl-CoA Delta(11) desaturase                                    </td><td>-7.277788</td><td>9.1423545</td><td>1.14000e-36</td><td>2.400000e-33</td></tr>\n",
       "\t<tr><td>LOC105190430</td><td>-8.367141</td><td> 6.5409884</td><td>4.810000e-51</td><td>8.690000e-48</td><td>acyl-CoA Delta(11) desaturase                                    </td><td>-8.864538</td><td>6.5357242</td><td>3.21000e-39</td><td>1.020000e-35</td></tr>\n",
       "\t<tr><td>LOC105182047</td><td>-7.854964</td><td> 2.9948256</td><td>3.410000e-23</td><td>1.720000e-20</td><td>elongation of very long chain fatty acids protein AAEL008004     </td><td>-6.689554</td><td>2.9650885</td><td>1.98000e-19</td><td>1.930000e-16</td></tr>\n",
       "\t<tr><td>LOC105182544</td><td>-5.977819</td><td> 8.2981666</td><td>5.840000e-16</td><td>1.230000e-13</td><td>fatty acid synthase                                              </td><td>       NA</td><td>       NA</td><td>         NA</td><td>          NA</td></tr>\n",
       "\t<tr><td>LOC105185296</td><td>-4.198066</td><td> 1.1288395</td><td>8.870000e-10</td><td>3.880000e-08</td><td>elongation of very long chain fatty acids protein 7              </td><td>-3.813467</td><td>1.0640692</td><td>3.72000e-07</td><td>1.340000e-05</td></tr>\n",
       "\t<tr><td>LOC112590320</td><td>-4.002081</td><td> 7.1111678</td><td>4.160000e-09</td><td>1.470000e-07</td><td>fatty acid synthase-like isoform X1                              </td><td>-4.272346</td><td>7.0937949</td><td>8.17000e-08</td><td>3.840000e-06</td></tr>\n",
       "\t<tr><td>LOC105180907</td><td>-3.996957</td><td> 6.0816650</td><td>5.510000e-09</td><td>1.890000e-07</td><td>acyl-CoA Delta(11) desaturase isoform X2                         </td><td>-4.608542</td><td>6.0474526</td><td>1.01000e-09</td><td>1.030000e-07</td></tr>\n",
       "\t<tr><td>LOC105185168</td><td>-1.964630</td><td> 9.5122145</td><td>7.840000e-05</td><td>5.341220e-04</td><td>fatty acid synthase                                              </td><td>       NA</td><td>       NA</td><td>         NA</td><td>          NA</td></tr>\n",
       "\t<tr><td>LOC112590305</td><td>-4.139110</td><td> 0.5728714</td><td>1.597420e-04</td><td>9.388660e-04</td><td>fatty acid synthase-like                                         </td><td>-3.561177</td><td>0.4730043</td><td>1.16370e-04</td><td>1.261819e-03</td></tr>\n",
       "\t<tr><td>LOC105186441</td><td>-2.846044</td><td> 7.2347044</td><td>3.309670e-04</td><td>1.637240e-03</td><td>fatty acid synthase-like                                         </td><td>-8.216428</td><td>7.0494603</td><td>1.29000e-41</td><td>5.430000e-38</td></tr>\n",
       "\t<tr><td>LOC105180672</td><td>-2.986761</td><td> 9.5887926</td><td>6.896250e-04</td><td>2.880991e-03</td><td>fatty acid synthase                                              </td><td>-7.382810</td><td>9.4255614</td><td>2.33000e-21</td><td>2.950000e-18</td></tr>\n",
       "\t<tr><td>LOC112590548</td><td>-3.712446</td><td>-0.2990144</td><td>2.010479e-03</td><td>6.673819e-03</td><td>elongation of very long chain fatty acids protein AAEL008004-like</td><td>       NA</td><td>       NA</td><td>         NA</td><td>          NA</td></tr>\n",
       "\t<tr><td>LOC112588737</td><td>       NA</td><td>        NA</td><td>          NA</td><td>          NA</td><td>fatty acid synthase-like                                         </td><td>-9.051802</td><td>7.0899687</td><td>1.36000e-43</td><td>1.720000e-39</td></tr>\n",
       "\t<tr><td>LOC112590175</td><td>       NA</td><td>        NA</td><td>          NA</td><td>          NA</td><td>fatty acid synthase-like                                         </td><td>-5.271302</td><td>5.6194296</td><td>9.93000e-09</td><td>7.090000e-07</td></tr>\n",
       "\t<tr><td>LOC105183865</td><td>       NA</td><td>        NA</td><td>          NA</td><td>          NA</td><td>fatty acid synthase-like isoform X1                              </td><td>-3.836275</td><td>5.2417284</td><td>1.31000e-08</td><td>8.720000e-07</td></tr>\n",
       "\t<tr><td>LOC105191892</td><td>       NA</td><td>        NA</td><td>          NA</td><td>          NA</td><td>LOW QUALITY PROTEIN: fatty acid synthase-like                    </td><td>-2.843582</td><td>8.0037357</td><td>1.52678e-04</td><td>1.527263e-03</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 17 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " genes & logFC\\_W\\_vs\\_G & logCPM\\_W\\_vs\\_G & PValue\\_W\\_vs\\_G & FDR\\_W\\_vs\\_G & Annotation & logFC\\_R\\_vs\\_G & logCPM\\_R\\_vs\\_G & PValue\\_R\\_vs\\_G & FDR\\_R\\_vs\\_G\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t LOC105191040 & -9.456765 &  8.6288891 & 2.190000e-73 & 2.780000e-69 & fatty acid synthase                                               &        NA &        NA &          NA &           NA\\\\\n",
       "\t LOC105191801 & -8.094335 &  9.1392166 & 5.170000e-53 & 1.310000e-49 & acyl-CoA Delta(11) desaturase                                     & -7.277788 & 9.1423545 & 1.14000e-36 & 2.400000e-33\\\\\n",
       "\t LOC105190430 & -8.367141 &  6.5409884 & 4.810000e-51 & 8.690000e-48 & acyl-CoA Delta(11) desaturase                                     & -8.864538 & 6.5357242 & 3.21000e-39 & 1.020000e-35\\\\\n",
       "\t LOC105182047 & -7.854964 &  2.9948256 & 3.410000e-23 & 1.720000e-20 & elongation of very long chain fatty acids protein AAEL008004      & -6.689554 & 2.9650885 & 1.98000e-19 & 1.930000e-16\\\\\n",
       "\t LOC105182544 & -5.977819 &  8.2981666 & 5.840000e-16 & 1.230000e-13 & fatty acid synthase                                               &        NA &        NA &          NA &           NA\\\\\n",
       "\t LOC105185296 & -4.198066 &  1.1288395 & 8.870000e-10 & 3.880000e-08 & elongation of very long chain fatty acids protein 7               & -3.813467 & 1.0640692 & 3.72000e-07 & 1.340000e-05\\\\\n",
       "\t LOC112590320 & -4.002081 &  7.1111678 & 4.160000e-09 & 1.470000e-07 & fatty acid synthase-like isoform X1                               & -4.272346 & 7.0937949 & 8.17000e-08 & 3.840000e-06\\\\\n",
       "\t LOC105180907 & -3.996957 &  6.0816650 & 5.510000e-09 & 1.890000e-07 & acyl-CoA Delta(11) desaturase isoform X2                          & -4.608542 & 6.0474526 & 1.01000e-09 & 1.030000e-07\\\\\n",
       "\t LOC105185168 & -1.964630 &  9.5122145 & 7.840000e-05 & 5.341220e-04 & fatty acid synthase                                               &        NA &        NA &          NA &           NA\\\\\n",
       "\t LOC112590305 & -4.139110 &  0.5728714 & 1.597420e-04 & 9.388660e-04 & fatty acid synthase-like                                          & -3.561177 & 0.4730043 & 1.16370e-04 & 1.261819e-03\\\\\n",
       "\t LOC105186441 & -2.846044 &  7.2347044 & 3.309670e-04 & 1.637240e-03 & fatty acid synthase-like                                          & -8.216428 & 7.0494603 & 1.29000e-41 & 5.430000e-38\\\\\n",
       "\t LOC105180672 & -2.986761 &  9.5887926 & 6.896250e-04 & 2.880991e-03 & fatty acid synthase                                               & -7.382810 & 9.4255614 & 2.33000e-21 & 2.950000e-18\\\\\n",
       "\t LOC112590548 & -3.712446 & -0.2990144 & 2.010479e-03 & 6.673819e-03 & elongation of very long chain fatty acids protein AAEL008004-like &        NA &        NA &          NA &           NA\\\\\n",
       "\t LOC112588737 &        NA &         NA &           NA &           NA & fatty acid synthase-like                                          & -9.051802 & 7.0899687 & 1.36000e-43 & 1.720000e-39\\\\\n",
       "\t LOC112590175 &        NA &         NA &           NA &           NA & fatty acid synthase-like                                          & -5.271302 & 5.6194296 & 9.93000e-09 & 7.090000e-07\\\\\n",
       "\t LOC105183865 &        NA &         NA &           NA &           NA & fatty acid synthase-like isoform X1                               & -3.836275 & 5.2417284 & 1.31000e-08 & 8.720000e-07\\\\\n",
       "\t LOC105191892 &        NA &         NA &           NA &           NA & LOW QUALITY PROTEIN: fatty acid synthase-like                     & -2.843582 & 8.0037357 & 1.52678e-04 & 1.527263e-03\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 17 × 10\n",
       "\n",
       "| genes &lt;chr&gt; | logFC_W_vs_G &lt;dbl&gt; | logCPM_W_vs_G &lt;dbl&gt; | PValue_W_vs_G &lt;dbl&gt; | FDR_W_vs_G &lt;dbl&gt; | Annotation &lt;chr&gt; | logFC_R_vs_G &lt;dbl&gt; | logCPM_R_vs_G &lt;dbl&gt; | PValue_R_vs_G &lt;dbl&gt; | FDR_R_vs_G &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| LOC105191040 | -9.456765 |  8.6288891 | 2.190000e-73 | 2.780000e-69 | fatty acid synthase                                               |        NA |        NA |          NA |           NA |\n",
       "| LOC105191801 | -8.094335 |  9.1392166 | 5.170000e-53 | 1.310000e-49 | acyl-CoA Delta(11) desaturase                                     | -7.277788 | 9.1423545 | 1.14000e-36 | 2.400000e-33 |\n",
       "| LOC105190430 | -8.367141 |  6.5409884 | 4.810000e-51 | 8.690000e-48 | acyl-CoA Delta(11) desaturase                                     | -8.864538 | 6.5357242 | 3.21000e-39 | 1.020000e-35 |\n",
       "| LOC105182047 | -7.854964 |  2.9948256 | 3.410000e-23 | 1.720000e-20 | elongation of very long chain fatty acids protein AAEL008004      | -6.689554 | 2.9650885 | 1.98000e-19 | 1.930000e-16 |\n",
       "| LOC105182544 | -5.977819 |  8.2981666 | 5.840000e-16 | 1.230000e-13 | fatty acid synthase                                               |        NA |        NA |          NA |           NA |\n",
       "| LOC105185296 | -4.198066 |  1.1288395 | 8.870000e-10 | 3.880000e-08 | elongation of very long chain fatty acids protein 7               | -3.813467 | 1.0640692 | 3.72000e-07 | 1.340000e-05 |\n",
       "| LOC112590320 | -4.002081 |  7.1111678 | 4.160000e-09 | 1.470000e-07 | fatty acid synthase-like isoform X1                               | -4.272346 | 7.0937949 | 8.17000e-08 | 3.840000e-06 |\n",
       "| LOC105180907 | -3.996957 |  6.0816650 | 5.510000e-09 | 1.890000e-07 | acyl-CoA Delta(11) desaturase isoform X2                          | -4.608542 | 6.0474526 | 1.01000e-09 | 1.030000e-07 |\n",
       "| LOC105185168 | -1.964630 |  9.5122145 | 7.840000e-05 | 5.341220e-04 | fatty acid synthase                                               |        NA |        NA |          NA |           NA |\n",
       "| LOC112590305 | -4.139110 |  0.5728714 | 1.597420e-04 | 9.388660e-04 | fatty acid synthase-like                                          | -3.561177 | 0.4730043 | 1.16370e-04 | 1.261819e-03 |\n",
       "| LOC105186441 | -2.846044 |  7.2347044 | 3.309670e-04 | 1.637240e-03 | fatty acid synthase-like                                          | -8.216428 | 7.0494603 | 1.29000e-41 | 5.430000e-38 |\n",
       "| LOC105180672 | -2.986761 |  9.5887926 | 6.896250e-04 | 2.880991e-03 | fatty acid synthase                                               | -7.382810 | 9.4255614 | 2.33000e-21 | 2.950000e-18 |\n",
       "| LOC112590548 | -3.712446 | -0.2990144 | 2.010479e-03 | 6.673819e-03 | elongation of very long chain fatty acids protein AAEL008004-like |        NA |        NA |          NA |           NA |\n",
       "| LOC112588737 |        NA |         NA |           NA |           NA | fatty acid synthase-like                                          | -9.051802 | 7.0899687 | 1.36000e-43 | 1.720000e-39 |\n",
       "| LOC112590175 |        NA |         NA |           NA |           NA | fatty acid synthase-like                                          | -5.271302 | 5.6194296 | 9.93000e-09 | 7.090000e-07 |\n",
       "| LOC105183865 |        NA |         NA |           NA |           NA | fatty acid synthase-like isoform X1                               | -3.836275 | 5.2417284 | 1.31000e-08 | 8.720000e-07 |\n",
       "| LOC105191892 |        NA |         NA |           NA |           NA | LOW QUALITY PROTEIN: fatty acid synthase-like                     | -2.843582 | 8.0037357 | 1.52678e-04 | 1.527263e-03 |\n",
       "\n"
      ],
      "text/plain": [
       "   genes        logFC_W_vs_G logCPM_W_vs_G PValue_W_vs_G FDR_W_vs_G  \n",
       "1  LOC105191040 -9.456765     8.6288891    2.190000e-73  2.780000e-69\n",
       "2  LOC105191801 -8.094335     9.1392166    5.170000e-53  1.310000e-49\n",
       "3  LOC105190430 -8.367141     6.5409884    4.810000e-51  8.690000e-48\n",
       "4  LOC105182047 -7.854964     2.9948256    3.410000e-23  1.720000e-20\n",
       "5  LOC105182544 -5.977819     8.2981666    5.840000e-16  1.230000e-13\n",
       "6  LOC105185296 -4.198066     1.1288395    8.870000e-10  3.880000e-08\n",
       "7  LOC112590320 -4.002081     7.1111678    4.160000e-09  1.470000e-07\n",
       "8  LOC105180907 -3.996957     6.0816650    5.510000e-09  1.890000e-07\n",
       "9  LOC105185168 -1.964630     9.5122145    7.840000e-05  5.341220e-04\n",
       "10 LOC112590305 -4.139110     0.5728714    1.597420e-04  9.388660e-04\n",
       "11 LOC105186441 -2.846044     7.2347044    3.309670e-04  1.637240e-03\n",
       "12 LOC105180672 -2.986761     9.5887926    6.896250e-04  2.880991e-03\n",
       "13 LOC112590548 -3.712446    -0.2990144    2.010479e-03  6.673819e-03\n",
       "14 LOC112588737        NA            NA              NA            NA\n",
       "15 LOC112590175        NA            NA              NA            NA\n",
       "16 LOC105183865        NA            NA              NA            NA\n",
       "17 LOC105191892        NA            NA              NA            NA\n",
       "   Annotation                                                       \n",
       "1  fatty acid synthase                                              \n",
       "2  acyl-CoA Delta(11) desaturase                                    \n",
       "3  acyl-CoA Delta(11) desaturase                                    \n",
       "4  elongation of very long chain fatty acids protein AAEL008004     \n",
       "5  fatty acid synthase                                              \n",
       "6  elongation of very long chain fatty acids protein 7              \n",
       "7  fatty acid synthase-like isoform X1                              \n",
       "8  acyl-CoA Delta(11) desaturase isoform X2                         \n",
       "9  fatty acid synthase                                              \n",
       "10 fatty acid synthase-like                                         \n",
       "11 fatty acid synthase-like                                         \n",
       "12 fatty acid synthase                                              \n",
       "13 elongation of very long chain fatty acids protein AAEL008004-like\n",
       "14 fatty acid synthase-like                                         \n",
       "15 fatty acid synthase-like                                         \n",
       "16 fatty acid synthase-like isoform X1                              \n",
       "17 LOW QUALITY PROTEIN: fatty acid synthase-like                    \n",
       "   logFC_R_vs_G logCPM_R_vs_G PValue_R_vs_G FDR_R_vs_G  \n",
       "1         NA           NA              NA             NA\n",
       "2  -7.277788    9.1423545     1.14000e-36   2.400000e-33\n",
       "3  -8.864538    6.5357242     3.21000e-39   1.020000e-35\n",
       "4  -6.689554    2.9650885     1.98000e-19   1.930000e-16\n",
       "5         NA           NA              NA             NA\n",
       "6  -3.813467    1.0640692     3.72000e-07   1.340000e-05\n",
       "7  -4.272346    7.0937949     8.17000e-08   3.840000e-06\n",
       "8  -4.608542    6.0474526     1.01000e-09   1.030000e-07\n",
       "9         NA           NA              NA             NA\n",
       "10 -3.561177    0.4730043     1.16370e-04   1.261819e-03\n",
       "11 -8.216428    7.0494603     1.29000e-41   5.430000e-38\n",
       "12 -7.382810    9.4255614     2.33000e-21   2.950000e-18\n",
       "13        NA           NA              NA             NA\n",
       "14 -9.051802    7.0899687     1.36000e-43   1.720000e-39\n",
       "15 -5.271302    5.6194296     9.93000e-09   7.090000e-07\n",
       "16 -3.836275    5.2417284     1.31000e-08   8.720000e-07\n",
       "17 -2.843582    8.0037357     1.52678e-04   1.527263e-03"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# full join the two lists and rename the columns\n",
    "Fb_joined_FA_genes <- full_join(Fb_W_vs_G_FA_genes,\n",
    "                                Fb_G_vs_R_FA_genes,\n",
    "                                by=c('Annotation','genes'))\n",
    "\n",
    "colnames(Fb_joined_FA_genes) <- sub(pattern='.x',\n",
    "                                    replacement='_W_vs_G',\n",
    "                                    x=colnames(Fb_joined_FA_genes))\n",
    "\n",
    "colnames(Fb_joined_FA_genes) <- sub(pattern='.y',\n",
    "                                    replacement='_R_vs_G',\n",
    "                                    x=colnames(Fb_joined_FA_genes))\n",
    "\n",
    "Fb_joined_FA_genes # check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 366,
   "id": "142944dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(Fb_joined_FA_genes,\n",
    "            \"topGO_output/__FA_genes_DOWN_in_W_or_R.tsv\",\n",
    "            sep=\"\\t\",\n",
    "            row.names=F,\n",
    "            quote=F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec8f9cb9",
   "metadata": {},
   "source": [
    "## Intersect worker- and revertant-enriched GO-terms"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d94d6367",
   "metadata": {},
   "source": [
    "Define a function to intersect significantly enriched GO-terms between the worker vs gamergate and the revertant vs gamergate comparisons for a given tissue and direction of gene regulation and plot the corresponding p-values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "id": "9f187e11",
   "metadata": {},
   "outputs": [],
   "source": [
    "intersect_castes <- function(tissue, direction=\"UP\", g2go_mapping, terms_to_plot=10){\n",
    "    \n",
    "    # import the outputs of runTOPGO() (see above) run on the lists of DE genes in W vs G and R vs G comparison\n",
    "    # add a column with term rank (1 corresponds to the lowest p-value)\n",
    "    go_table_W <- read.table(paste0(\"topGO_output/\",\n",
    "                                    tissue,\n",
    "                                    \"_W_vs_G_\",\n",
    "                                    deparse(substitute(g2go_mapping)),\n",
    "                                    \"_\",\n",
    "                                    direction,\n",
    "                                    \".tsv\"),\n",
    "                             header=T,\n",
    "                             sep='\\t',\n",
    "                             quote='')\n",
    "    go_table_W$rank <- 1:nrow(go_table_W)\n",
    "\n",
    "    go_table_R <- read.table(paste0(\"topGO_output/\",\n",
    "                                    tissue,\n",
    "                                    \"_G_vs_R_\",\n",
    "                                    deparse(substitute(g2go_mapping)),\n",
    "                                    \"_\",\n",
    "                                    direction,\n",
    "                                    \".tsv\"),\n",
    "                             header=T,\n",
    "                             sep='\\t',\n",
    "                             quote='')\n",
    "    go_table_R$rank <- 1:nrow(go_table_R)\n",
    "    \n",
    "    # intersect GO-terms in the W vs G and R vs G tables,\n",
    "    # calculate average rank and sort by it,\n",
    "    # convert to long format\n",
    "    plot_df <- \n",
    "     go_table_W %>%\n",
    "     inner_join(go_table_R, by=c('GO.ID', 'Term')) %>%\n",
    "     rowwise() %>%\n",
    "     mutate(average_rank=round(mean(c(rank.x,rank.y)),0)) %>%\n",
    "     arrange(average_rank) %>%\n",
    "     select(GO.ID, Term, elimFisher.x, elimFisher.y, average_rank) %>%\n",
    "     pivot_longer(c(elimFisher.x,elimFisher.y), names_to=\"comparison\", values_to=\"pvalue\")\n",
    "    \n",
    "    # convert \"elimFisher.x\" in the comparison column to \"W_vs_G\"\n",
    "    # and \"elimFisher.y\" to \"R_vs_G\" (.x and .y names are generated by inner_join() above)\n",
    "    plot_df$comparison <- ifelse(test=plot_df$comparison==\"elimFisher.x\",\n",
    "                                 yes=\"W_vs_G\",\n",
    "                                 no=\"R_vs_G\")\n",
    "    \n",
    "    # select as many terms as specified in terms_to_plot unless there are fewer significant terms\n",
    "    # terms_to_plot needs to be multiplied by 2 because each term appears on 2 lines in plot_df (W_vs_G and R_vs_G)\n",
    "    if ( nrow(plot_df)>=terms_to_plot*2 ) {\n",
    "\n",
    "        # select terms_to_plot terms\n",
    "        plot_df <- plot_df[1:(terms_to_plot*2),]\n",
    "        # set plot height for later\n",
    "        plot_height <- terms_to_plot/1.5\n",
    "        \n",
    "        # add spaces in front of every term name such that the total number of characters equals 100\n",
    "        # (they are typically shorter than that)\n",
    "        # this is necessary to make different plots more comparable\n",
    "        for (i in 1:nrow(plot_df)) {\n",
    "            term <- as.character(plot_df$Term[i])\n",
    "            original_nchar <- nchar(term)\n",
    "            nchar_to_add <- 100-original_nchar\n",
    "            term <- paste0(paste(rep(' ', nchar_to_add), collapse=\"\"),\n",
    "                           term)\n",
    "            plot_df$Term[i] <- term\n",
    "        }\n",
    "\n",
    "    } else if ( nrow(plot_df)>0 ) {\n",
    "\n",
    "        # no selection is necessary if the number of overlapping GO-terms is smaller than terms_to_plot\n",
    "        plot_df <- plot_df\n",
    "        # set plot height for later\n",
    "        plot_height <- nrow(plot_df)/1.5\n",
    "\n",
    "        # add spaces in front of every term name such that the total number of characters equals 100\n",
    "        # (they are typically shorter than that)\n",
    "        # this is necessary to make different plots more comparable\n",
    "        for (i in 1:nrow(plot_df)) {\n",
    "            term <- as.character(plot_df$Term[i])\n",
    "            original_nchar <- nchar(term)\n",
    "            nchar_to_add <- 100-original_nchar\n",
    "            term <- paste0(paste(rep(' ', nchar_to_add), collapse=\"\"),\n",
    "                           term)\n",
    "            plot_df$Term[i] <- term\n",
    "        }\n",
    "    \n",
    "    } else {\n",
    "        \n",
    "        # no selection is necessary if the number of overlapping GO-terms is 0\n",
    "        plot_df <- plot_df\n",
    "        # set an arbitrarily small plot height for later\n",
    "        plot_height <- 1\n",
    "        # print a warning that there are no overlapping GO-terms\n",
    "        warning(paste0(\"NO OVERLAP BETWEEN GO-TERMS IN \",\n",
    "                       tissue,\n",
    "                       \"_W_vs_G_\",\n",
    "                       deparse(substitute(g2go_mapping)),\n",
    "                       \"_\",\n",
    "                       direction,\n",
    "                       \".tsv\",\n",
    "                       \" AND \",\n",
    "                       tissue,\n",
    "                       \"_G_vs_R_\",\n",
    "                       deparse(substitute(g2go_mapping)),\n",
    "                       \"_\",\n",
    "                       direction,\n",
    "                       \".tsv\"))\n",
    "    }\n",
    "    \n",
    "    # plot -log(p-value) in each comparison\n",
    "    plot_df$Term <- factor(plot_df$Term,\n",
    "                           levels=rev(unique(plot_df$Term)))\n",
    "\n",
    "    p1 <- ggplot(data=plot_df,\n",
    "             mapping=aes(x=Term,\n",
    "                         y=-log(pvalue),\n",
    "                         fill=comparison)) +\n",
    "        geom_col(position=\"dodge\",\n",
    "                 width=0.75) +\n",
    "        coord_flip() +\n",
    "        ylim(0,36) +\n",
    "        scale_fill_manual(values=c(\"#0072B2\", \"#D55E00\")) +\n",
    "        theme(text=element_text(size=24))+\n",
    "        ggtitle(paste0(tissue,\n",
    "                       \"_\",\n",
    "                       direction))\n",
    "\n",
    "    filename <- paste0(\"topGO_output/_\",\n",
    "                       tissue,\n",
    "                       \"_\",\n",
    "                       deparse(substitute(g2go_mapping)),\n",
    "                       \"_\",\n",
    "                       direction,\n",
    "                       \".eps\")\n",
    "    ggsave(filename,\n",
    "           p1,\n",
    "           width=15,\n",
    "           height=plot_height) \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "id": "a14a6394",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in intersect_castes(\"cBr\", \"DOWN\", gene2go_Karl, 20):\n",
      "“NO OVERLAP BETWEEN GO-TERMS IN cBr_W_vs_G_gene2go_Karl_DOWN.tsv AND cBr_G_vs_R_gene2go_Karl_DOWN.tsv”\n",
      "Warning message in intersect_castes(\"cBr\", \"DOWN\", gene2go_CrowdGO, 20):\n",
      "“NO OVERLAP BETWEEN GO-TERMS IN cBr_W_vs_G_gene2go_CrowdGO_DOWN.tsv AND cBr_G_vs_R_gene2go_CrowdGO_DOWN.tsv”\n"
     ]
    }
   ],
   "source": [
    "intersect_castes(tissue=\"Fb\",\n",
    "                 direction=\"UP\",\n",
    "                 g2go_mapping=gene2go_Karl,\n",
    "                 terms_to_plot=20)\n",
    "intersect_castes(\"Fb\", \"UP\", gene2go_CrowdGO, 20)\n",
    "intersect_castes(\"Fb\", \"DOWN\", gene2go_Karl, 20)\n",
    "intersect_castes(\"Fb\", \"DOWN\", gene2go_CrowdGO, 20)\n",
    "intersect_castes(\"Ov\", \"UP\", gene2go_Karl, 20)\n",
    "intersect_castes(\"Ov\", \"UP\", gene2go_CrowdGO, 20)\n",
    "intersect_castes(\"Ov\", \"DOWN\", gene2go_Karl, 20)\n",
    "intersect_castes(\"Ov\", \"DOWN\", gene2go_CrowdGO, 20)\n",
    "intersect_castes(\"cBr\", \"UP\", gene2go_Karl, 20)\n",
    "intersect_castes(\"cBr\", \"UP\", gene2go_CrowdGO, 20)\n",
    "intersect_castes(\"cBr\", \"DOWN\", gene2go_Karl, 20)\n",
    "intersect_castes(\"cBr\", \"DOWN\", gene2go_CrowdGO, 20)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
